// @arcteryx/design-system/src/components/molecules/fullWidthCard/__mocks__/fullWidthCardMocks.ts
import { type RichTextBlocks } from "../../richText/RichText.types";
import { type FullWidthCardProps } from "../FullWidthCard.types";

const fullWidthCardRichText: RichTextBlocks = [
  {
    _key: "fd259cce9afa",
    _type: "block",
    style: "h3",
    markDefs: [],
    children: [
      {
        _key: "cc33163cbdf40",
        _type: "span",
        marks: [],
        text: "RETAIL AND REGIONAL HUB JOBS",
      },
    ],
  },
  {
    _key: "536d411dbccf",
    _type: "block",
    children: [
      {
        _key: "fe42f34f528d0",
        _type: "span",
        marks: [],
        text: "Our global retail spaces invite guests to connect with the brand and become part of a community. Regional hubs are key locations for the brand, providing signature experiences, partnerships and next-level community offerings.",
      },
    ],
    markDefs: [],
    style: "normal",
  },
  {
    _key: "b21132af21a3",
    _type: "block",
    children: [
      {
        _key: "0bcf461b5ea5",
        _type: "span",
        marks: [],
        text: "",
      },
    ],
    markDefs: [],
    style: "normal",
  },
  {
    _type: "atLink",
    label: "View Jobs",
    linkType: "default",
    shouldDisplayAsButton: true,
    url: "https://www.google.com",
    variant: "Standard",
  },
];

export const fullWidthCardData: FullWidthCardProps = {
  image: {
    src: "https://cdn.sanity.io/images/inkbj32c/development/a82519c8ecc2775aff855096eefe9cbc314d3744-1060x754.png?auto=format&q=75",
    alt: "card image",
  },
  content: fullWidthCardRichText,
  textAlign: "left",
  imageVariant: "default",
};

export const fullWidthCardWithNoText: FullWidthCardProps = {
  image: {
    src: "https://cdn.sanity.io/images/inkbj32c/development/a82519c8ecc2775aff855096eefe9cbc314d3744-1060x754.png?auto=format&q=75",
    alt: "card image",
  },
  content: null,
  textAlign: "left",
  imageVariant: "square",
};
