// @arcteryx/components-content-blocks/src/components/backgrounds/HeroBackgroundImage.jsx
import React from "react";
import { breakpoints } from "../styles/constants";
import { generateSource } from "./helpers";
import { BackgroundImage } from "./BackgroundImage";
// The Width and Height are pulled from the Design System. The values are the height and width at each breakpoint.


export const HeroBackgroundImage = ({
  src,
  mobileSmall,
  mobileLarge,
  tablet,
  desktopSmall,
  desktop,
  desktopLarge,
  title,
}) => {
  const sources = [
    generateSource({
      width: 1920,
      height: 1085,
      media: `(min-width: ${breakpoints.desktopLarge}px)`,
      sizes: "100vw",
      focalPointX: desktopLarge?.focalPointX,
      focalPointY: desktopLarge?.focalPointY,
    }),
    generateSource({
      width: 1920,
      height: 678,
      media: `(min-width: ${breakpoints.desktop}px)`,
      sizes: "100vw",
      focalPointX: desktop?.focalPointX,
      focalPointY: desktop?.focalPointY,
    }),
    generateSource({
      width: 992,
      height: 600,
      media: `(min-width: ${breakpoints.desktopSmall}px)`,
      sizes: "100vw",
      focalPointX: desktopSmall?.focalPointX,
      focalPointY: desktopSmall?.focalPointY,
    }),
    generateSource({
      width: 768,
      height: 600,
      media: `(min-width: ${breakpoints.tablet}px)`,
      sizes: "100vw",
      focalPointX: tablet?.focalPointX,
      focalPointY: tablet?.focalPointY,
    }),
    generateSource({
      width: 576,
      height: 600,
      media: `(min-width: ${breakpoints.mobileLarge}px)`,
      sizes: "100vw",
      focalPointX: mobileLarge?.focalPointX,
      focalPointY: mobileLarge?.focalPointY,
    }),
    generateSource({
      width: 375,
      height: 561,
      media: `(min-width: ${breakpoints.mobileSmall}px)`,
      sizes: "100vw",
      focalPointX: mobileSmall?.focalPointX,
      focalPointY: mobileSmall?.focalPointY,
    }),
  ];
  return <BackgroundImage src={src} sources={sources} shouldLazyLoad={false} sizes="100vw" title={title} />;
};
