// @arcteryx/components-content-blocks/src/components/backgrounds/SecondaryBackgroundImage.jsx
import React from "react";
import { BackgroundImage } from "./BackgroundImage";
import { generateSource } from "./helpers";
import { breakpoints } from "../styles/constants";
// The Width and Height are pulled from the Design System. The values are the height and width at each breakpoint.

export const SecondaryBackgroundImage = ({
  src,
  mobileSmall,
  mobileLarge,
  tablet,
  desktopSmall,
  desktop,
  desktopLarge,
}) => {
  const sources = [
    generateSource({
      width: 952.5,
      height: 696,
      media: `(min-width: ${breakpoints.desktopLarge}px)`,
      sizes: "50vw",
      focalPointX: desktopLarge?.focalPointX,
      focalPointY: desktopLarge?.focalPointY,
    }),
    generateSource({
      width: 591.66,
      height: 430.09,
      media: `(min-width: ${breakpoints.desktop}px)`,
      sizes: "50vw",
      focalPointX: desktop?.focalPointX,
      focalPointY: desktop?.focalPointY,
    }),
    generateSource({
      width: 487.66,
      height: 359,
      media: `(min-width: ${breakpoints.desktopSmall}px)`,
      sizes: "50vw",
      focalPointX: desktopSmall?.focalPointX,
      focalPointY: desktopSmall?.focalPointY,
    }),
    generateSource({
      width: 425,
      height: 425,
      media: `(min-width: ${breakpoints.tablet}px)`,
      sizes: "50vw",
      focalPointX: tablet?.focalPointX,
      focalPointY: tablet?.focalPointY,
    }),
    generateSource({
      width: 576,
      height: 576,
      media: `(min-width: ${breakpoints.mobileLarge}px)`,
      sizes: "100vw",
      focalPointX: mobileLarge?.focalPointX,
      focalPointY: mobileLarge?.focalPointY,
    }),
    generateSource({
      width: 320,
      height: 320,
      media: `(min-width: ${breakpoints.mobileSmall}px)`,
      sizes: "100vw",
      focalPointX: mobileSmall?.focalPointX,
      focalPointY: mobileSmall?.focalPointY,
    }),
  ];

  return <BackgroundImage src={src} sources={sources} shouldLazyLoad={true} sizes="(min-width:768px) 100vw, 50vw" />;
};
