// @arcteryx/components-content-blocks/src/components/backgrounds/VideoBackground.jsx
import React from "react";
import styled from "styled-components";

const StyledDiv = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
`;

const StyledVideo = styled.video`
  position: absolute;
  width: 100%;
  height: 100%;
  object-fit: cover;
  z-index: 1;
`;

/*
 * https://en.wikipedia.org/wiki/MPEG-4_Part_14
 * https://en.wikipedia.org/wiki/WebM
 * https://en.wikipedia.org/wiki/Ogg
 */

const videoFormats = {
  mp4: "video/mp4",
  m4a: "video/mp4",
  m4p: "video/mp4",
  m4b: "video/mp4",
  m4r: "video/mp4",
  m4v: "video/mp4",
  webm: "video/webm",
  ogg: "video/ogg",
  ogv: "video/ogg",
  oga: "video/ogg",
  ogx: "video/ogg",
  ogm: "video/ogg",
  spx: "video/ogg",
  opus: "video/ogg",
};

export const VideoBackground = ({ url, children }) => {
  return (
    <StyledDiv>
      <>
        <StyledVideo autoPlay={true} loop={true} muted={true} playsInline={true}>
          <source src={url} type={videoFormats[url.split(".").pop()]} />
        </StyledVideo>
        {children}
      </>
    </StyledDiv>
  );
};
