// @arcteryx/components-accordion/src/Accordion.stories.js
import React from "react";
import { Accordion, RadioAccordion } from "./index";
import { H4, P } from "@arcteryx/components-typography";
import { Icon } from "@arcteryx/components-icons";

export default {
  title: "Accordion",
  component: Accordion,
};

export const Accordions = () => {
  return (
    <>
      <Accordion title="H1 - Title Header" headingLevel="1" isOpenOnPageLoad={true}>
        <H4>Technical Features</H4>
        <P>
          Due to an unanticipated change during battery production, we are revising the minimum operating temperature
          limit for Voltair Airbags from -30°C (-22°F) to -20°C (-4°F). No other parts of the Voltair Airbag have been
          affected by this change. <a href="/">Focusable link</a>
        </P>
      </Accordion>

      <Accordion title="H2 - Title Header" headingLevel="2">
        <P>
          Due to an unanticipated change during battery production, we are revising the minimum operating temperature
          limit for Voltair Airbags from -30°C (-22°F) to -20°C (-4°F). No other parts of the Voltair Airbag have been
          affected by this change.
        </P>
      </Accordion>
      <Accordion title="H3 - Title Header" headingLevel="3">
        <P>
          Due to an unanticipated change during battery production, we are revising the minimum operating temperature
          limit for Voltair Airbags from -30°C (-22°F) to -20°C (-4°F). No other parts of the Voltair Airbag have been
          affected by this change.
        </P>
      </Accordion>
      <Accordion
        title="H3 - Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        headingLevel="3"
      >
        <P>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
          magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
          consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
          pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id
          est laborum.
        </P>
      </Accordion>
      <Accordion title="H4 - Title Header" headingLevel="4">
        <P>
          Due to an unanticipated change during battery production, we are revising the minimum operating temperature
          limit for Voltair Airbags from -30°C (-22°F) to -20°C (-4°F). No other parts of the Voltair Airbag have been
          affected by this change.
        </P>
      </Accordion>
      <Accordion title="P - Title Header" headingLevel="5">
        <P>
          Due to an unanticipated change during battery production, we are revising the minimum operating temperature
          limit for Voltair Airbags from -30°C (-22°F) to -20°C (-4°F). No other parts of the Voltair Airbag have been
          affected by this change.
        </P>
      </Accordion>
    </>
  );
};

const ExampleForm = ({ onSubmit }) => {
  return (
    <form onSubmit={(e) => {
      e.preventDefault();
      onSubmit("FormData");
    }}>

      <p>Your form elements here</p>
      <button type="submit">Submit</button>
    </form>
  );
};

const ExampleTitle = () => {
  return (
  <div style={{ display: 'flex', alignItems: 'center' }}>
    <Icon type={"visa-logo"} width={"50"} height={"30"}/>
    <span>Credit or Debit Card</span>
  </div>
  );
}


export const RadioVariant = () => {
  const items = [
    {
      title: <ExampleTitle />,
      content: <ExampleForm onSubmit={(data) => console.log(data)} />,
      id: "form1",
    },
    {
      title: <ExampleTitle />,
      content: <ExampleForm onSubmit={(data) => console.log(data)} />,
      id: "form2",
    },
    {
      title: <ExampleTitle />,
      content: <ExampleForm onSubmit={(data) => console.log(data)} />,
      id: "form3",
    },
  ]

  return (
    <div>
      <RadioAccordion items={items} />
    </div>
  );
}
