// @arcteryx/components-tabs/src/components/TabContainer.js
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { H1, H3 } from "@arcteryx/components-typography";

const TabDiv = styled.div`
  display: ${({ isSelected }) => (isSelected ? "block" : "none")};
`;

const StyledTitle = styled(H1)`
  font-size: 1.1rem;
  text-align: left;
  margin-top: 2rem;
  margin-bottom: 0.5rem;
`;

const StyledCategory = styled(H3)`
  margin-bottom: 1rem;
  font-size: 1.5rem;
  text-align: left;
`

const TabContainer = ({ title, category, children, selectedIndex, index, isConditionalRenderOff = false }) => {
  const isSelected = selectedIndex === index;
  const hasTitle = title?.length > 0 || false;
  const hasCategory = category?.length > 0 || false;

  if (isConditionalRenderOff) {
    return (
      <TabDiv data-testid={`tab-container-${index}`} key={`tab-container-${index}`} isSelected={isSelected}>
        {children}
      </TabDiv>
    );
  }
  return (
    <>
      {isSelected && (
        <div data-testid={`tab-container-${index}`} key={`tab-container-${index}`}>
          {hasTitle && (
            <StyledTitle>{title}</StyledTitle>
          )}
          {hasCategory && (
            <StyledCategory>{category}</StyledCategory>
          )}
          {children}
        </div>
      )}
    </>
  );
};
TabContainer.propTypes = {
  title: PropTypes.string,
  category: PropTypes.string,
  children: PropTypes.node,
  selectedIndex: PropTypes.number,
  index: PropTypes.number,
};
export default TabContainer;
