// @arcteryx/design-system/src/components/molecules/card/Card.stories.tsx
import { type Meta, type StoryObj } from "@storybook/react";
import { Card } from "./Card";
import {
  dynamicCardData,
  imageCardData,
  mensSizingChartCardData,
  shoeSizingChartCardData,
  videoCardData,
} from "./__mocks__/cardMocks";
import { longMeasurementBlocks } from "./__mocks__/cardRichText";

const meta: Meta<typeof Card> = {
  component: Card,
  title: "Design System/Molecules/Card",
};
export default meta;

type Story = StoryObj<typeof Card>;

export const SuperSimpleCard: Story = {
  args: { ...imageCardData },
};

export const SimpleCardModel: Story = {
  name: "Simple Card - Model",
  args: mensSizingChartCardData,
};

export const SimpleCardShoe: Story = {
  name: "Simple Card - Shoe",
  args: shoeSizingChartCardData,
};

export const SimpleCardVideo: Story = {
  args: videoCardData,
};

export const DynamicCardContentStrip: Story = {
  args: { ...dynamicCardData },
};

export const DynamicCardContentStripLong: Story = {
  args: { ...dynamicCardData, content: longMeasurementBlocks },
};
