// @arcteryx/design-system/src/components/molecules/richText/__mocks__/blocks.ts
import { type RichTextBlocks } from "../RichText.types";

export const buttonBlock: RichTextBlocks = [
  {
    _type: "atLink",
    shouldDisplayAsButton: true,
    variant: "Secondary-cta",
    label: "Shop",
    linkType: "default",
    url: "https://example.com",
  },
];

export const tableBlock: RichTextBlocks = [
  {
    _type: "table",
    rows: [
      {
        _type: "tableRow",
        cells: ["Table Header", "Table Header 2"],
      },
      {
        _type: "tableRow",
        cells: ["Table Body", "Table Body 2"],
      },
    ],
  },
];

export const spanBlock: RichTextBlocks = [
  {
    _key: "abc123B",
    _type: "block",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const h1Block: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "h1",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const h1AltBlock: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "h1Alt",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const h2Block: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "h2",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const h3Block: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "h3",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const h4Block: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "h4",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const h5Block: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "h5",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const h6Block: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "h6",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const SubheadingBlock: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "subheading",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const SubheadingAltBlock: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "subheadingAlt",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const MiceSmallBlock: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "miceSmall",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const MiceXSmallBlock: RichTextBlocks = [
  {
    _key: "abc123",
    _type: "block",
    style: "miceXSmall",
    children: [
      {
        _key: "def456",
        _type: "span",
        text: "Hello, world!",
      },
    ],
  },
];

export const richTextBlockList: RichTextBlocks = [...buttonBlock, ...spanBlock];
