// @arcteryx/components-breadcrumb/src/Breadcrumb.stories.js
import React from "react";
import { Breadcrumb } from "./index";

export default {
  title: "Breadcrumb",
  component: Breadcrumb,
};

export const Standard = () => {
  const plpEN = {
    title: "PLP Test: Outdoor - Women's Softshell Jackets",
    breadcrumbs: [
      { label: "Arc'teryx", url: "https://arcteryx.com/ca/en" },
      { label: "Women's", url: "https://arcteryx.com/ca/en/c/womens" },
      {
        label: "Shell Jackets",
        url: "https://arcteryx.com/ca/en/c/womens/shell-jackets",
      },
      {
        label: "Softshell",
        url: "https://arcteryx.com/ca/en/c/womens/shell-jackets/sub-cat=softshell",
      },
    ],
    market: "outdoor",
    tag: "h1",
  };
  return (
    <>
      <Breadcrumb
        breadcrumbs={plpEN.breadcrumbs}
        colour={plpEN.colour}
        theme={plpEN.theme}
        tag={plpEN.tag}
        isStorybook={true}
      />
    </>
  );
};

export const Inline = () => {
  const pdpInline = {
    title: "Inline PDP Test: Outdoor - Men's Alpah SV Jacket (Theme: inline)",
    breadcrumbs: [
      { label: "Arc'teryx", url: "/ca/en" },
      {
        label: "Men's",
        url: "/ca/en/c/mens",
      },
      {
        label: "Shell Jackets",
        url: "/ca/en/c/mens/shell-jackets",
      },
      {
        label: "Alpha SV Jacket",
        url: "/ca/en/shop/mens/alpha-sv-jacket",
      },
    ],
    theme: "inline",
    tag: "div",
  };
  return (
    <>
      <Breadcrumb
        breadcrumbs={pdpInline.breadcrumbs}
        colour={pdpInline.colour}
        theme={pdpInline.theme}
        tag={pdpInline.tag}
        isStorybook={true}
      />
    </>
  );
};
