// @arcteryx/design-system/src/components/molecules/tableSizing/TableSizing.stories.tsx
import { TableSizing } from "./TableSizing";
import React from "react";
import styled from "styled-components";
import { type Meta, type StoryObj } from "@storybook/react";
import {
  cmTable,
  footwearSizingTable,
  inchesTable,
  internationalTable,
} from "./__mocks__/tableSizingMocks";

const StyledContainer = styled.div`
  width: 100%;
  margin: 0 auto;
  padding: 1rem 0;
`;

interface InitialProps {
  descriptions: string[];
  table: string[][];
}

const meta: Meta = {
  title: "Design System / Molecules / Table Sizing",
  component: TableSizing,
};

export default meta;

type Story = StoryObj<any>;

const renderTable = (args: InitialProps): React.ReactElement => (
  <StyledContainer>
    <TableSizing description="" measuringType="" table={args?.table} />
  </StyledContainer>
);

export const SizingChartCentimeters: Story = {
  render: renderTable,
  args: cmTable,
};

export const SizingChartInches: Story = {
  render: renderTable,
  args: inchesTable,
};

export const SizingChartInternational: Story = {
  render: renderTable,
  args: internationalTable,
};

export const SizingChartShoes: Story = {
  render: renderTable,
  args: footwearSizingTable,
};
