// @arcteryx/design-system/src/components/molecules/fullWidthCard/FullWidthCard.test.tsx
import "@testing-library/jest-dom";
import React from "react";
import { render, screen } from "@testing-library/react";
import { FullWidthCard } from "./FullWidthCard";
import { type FullWidthCardProps } from "./FullWidthCard.types";
import { type RichTextBlocks } from "../richText";

const fullWidthCardRichText: RichTextBlocks = [
  {
    _key: "fd259cce9afa",
    _type: "block",
    style: "h3",
    markDefs: [],
    children: [
      {
        _key: "cc33163cbdf40",
        _type: "span",
        marks: [],
        text: "TEST HEADLINE",
      },
    ],
  },
  {
    _key: "536d411dbccf",
    _type: "block",
    children: [
      {
        _key: "fe42f34f528d0",
        _type: "span",
        marks: [],
        text: "TEST PARAGRAPH",
      },
    ],
    markDefs: [],
    style: "normal",
  },
  {
    _key: "b21132af21a3",
    _type: "block",
    children: [
      {
        _key: "0bcf461b5ea5",
        _type: "span",
        marks: [],
        text: "",
      },
    ],
    markDefs: [],
    style: "normal",
  },
  {
    _type: "atLink",
    label: "TEST LINK",
    linkType: "default",
    shouldDisplayAsButton: true,
    url: "https://www.arcteryx.com",
    variant: "Standard",
  },
];

export const fullWidthCardData: FullWidthCardProps = {
  image: {
    src: "https://cdn.sanity.io/images/inkbj32c/development/a82519c8ecc2775aff855096eefe9cbc314d3744-1060x754.png?auto=format&q=75",
    alt: "card image",
  },
  content: fullWidthCardRichText,
  textAlign: "left",
};

export const fullWidthCardWithNoTextData: FullWidthCardProps = {
  image: {
    src: "https://cdn.sanity.io/images/inkbj32c/development/a82519c8ecc2775aff855096eefe9cbc314d3744-1060x754.png?auto=format&q=75",
    alt: "card image",
  },
  content: null,
  textAlign: "left",
};

describe("FullWidthCard", () => {
  it("renders the content correctly", () => {
    render(<FullWidthCard {...fullWidthCardData} />);
    const contentElement = screen.getByText("TEST HEADLINE");

    expect(contentElement).toBeInTheDocument();
  });

  it("renders the image correctly", () => {
    render(<FullWidthCard {...fullWidthCardData} />);
    const imageElement = screen.getByAltText(
      fullWidthCardData.image?.alt ?? "",
    );
    expect(imageElement).toBeInTheDocument();
  });
  it("renders the image and no text", () => {
    const { container } = render(
      <FullWidthCard {...fullWidthCardWithNoTextData} />,
    );

    const imageElement = screen.getByAltText(
      fullWidthCardData.image?.alt ?? "",
    );

    expect(imageElement).toBeInTheDocument();
    expect(container.childNodes[0].childNodes.length).toBe(1);
  });
});
