// @arcteryx/components-content-blocks/src/ContentBlocks.stories.js
import React from "react";
import { CMSButton } from './components/buttons/CMSButton';


const storyWrapper = (storyFn) => (
  <div style={{ width: '100%', height: '100%', display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'center' }}>
    {storyFn()}
  </div>
);

export default {
  title: 'Content Blocks',
  parameters: {
    layout: 'fullscreen',
  },
  decorators: [storyWrapper],
};

const ButtonTemplate = (props) => <CMSButton {...props}>Hi </CMSButton>

export const SimpleButton = ButtonTemplate.bind({});
SimpleButton.args = { label: 'Shop Mens', color: 'white', backgroundColor: 'black' };
SimpleButton.storyName = 'Button'

export const ResponsiveButton = ButtonTemplate.bind({});
ResponsiveButton.args = {
  label: 'Shop Mens',
  color: 'white',
  backgroundColor: 'black',
  width: '140px',
  fonstSize:'15px',
  tabletStyles: { width: '200px', fontSize:'12px' },
  desktopStyles: { width: '300px', fontSize:'8px' }
};

export const ButtonAdornment = ButtonTemplate.bind({});
ButtonAdornment.args = {
  label: 'COLD WX LT HOODY GEN2 VIDEO',
  adornment: "$",
  color: 'white',
  backgroundColor: 'black',
  width: '300px',
};
