// @arcteryx/components-content-blocks/src/PageLayout.stories.js
import React from "react";
import { PageLayout } from ".";
import pageData from "../__tests__/mocks/outletPageData";
import designTemplate from "../__tests__/mocks/designTemplate";
import { UIHeader, UIFooter } from "@arcteryx/components-universal";
import { SiteContextProvider } from "@arcteryx/components-contexts";


const storyWrapper = (storyFn) => (
    <SiteContextProvider market="outdoor" country="ca" language="en">
        <UIHeader />
        {storyFn()}
        <UIFooter />
    </SiteContextProvider>
);

export default {
    title: 'Page Layout',
    parameters: {
        layout: 'fullscreen',
    },
    decorators: [storyWrapper],
};

 const PageLayoutTemplate = (props) =>     <PageLayout {...props} />

 export const DesignTemplate = PageLayoutTemplate.bind({});
 DesignTemplate.args = { sections: designTemplate.sections };
 DesignTemplate.storyName = 'Outdoor Design Template'

 export const PageTemplate = PageLayoutTemplate.bind({});
 PageTemplate.args = { sections: pageData.sections };
 PageTemplate.storyName = 'Outlet Preview'
