// @arcteryx/components-layouts/src/Layout.stories.js
import React from "react";
import { ContentLayout, HelpBottomLayout } from "../src/index";
import styled from "styled-components";

// const stories = storiesOf("Layouts", module);
const StoryWrapper = styled(ContentLayout)`
  padding: 1rem 0;
`;

// These styles are specific for Order History overriding the HTML that would come from the CMS
const OrderHistoryWrapper = styled.div`
  h2 {
    font-size: 1.25rem;
    margin: 0 0 1em 0;
    text-transform: uppercase;
  }
  a {
    color: var(--colour-black);
    text-decoration: underline;
  }
  a.replace-cs-phone {
    text-decoration: none;
  }

  @media screen and (min-width: 48em) {
    .help__icon-tooltip {
      padding-left: 3rem;
    }
    .help__icon-tooltip:after {
      content: url(//images.arcteryx.com/foundation-ui/svgs/Question_Icon.svg);
      position: absolute;
      left: 0;
      top: 0;
      width: 30px;
      height: 30px;
      display: block;
    }
  }
`;

const cmscontent = {
  path: "account/returns",
  left: [
    {
      content:
        '<h2>Returns Policy</h2>\n\n<ul><li>\n\t<p>Items purchased at <a href="https:///www.arcteryx.com">arcteryx.com</a> may be returned for up to 30 days <span lang="EN-US" xml:lang="EN-US" xml:lang="EN-US"><span><span><span><span>after the original delivery date</span></span></span></span></span>. Purchases from outlet.arcteryx.com must be returned on the <a href="https://outlet.arcteryx.com/help/returns">Outlet Returns page</a>.</p>\n\t</li>\n\t<li>\n\t<p>Items may also be brought to your nearest Brand Store where the team can facilitate your return back to our warehouse for processing.</p>\n\t</li>\n\t<li>\n\t<p>For safety and health reasons, climbing harnesses, underwear and the Reusable Face Mask cannot be returned. Shoes must be in their original packaging. The Voltair must not have been deployed.</p>\n\t</li>\n\t<li>\n\t<p>Refunds are issued in the form of the original payment. Arc\'teryx will provide a full refund for the value of the purchased item in the same form as the original payment.</p>\n\t</li>\n\t<li>\n\t<p>Refunds may take between 5-7 business days to process once an item has been delivered to our warehouse. *For United Kingdom returns - Currently, refunds may take up to 3 weeks to be processed once received.</p>\n\t</li>\n\t<li>\n\t<p>Items must be in as new condition and returned in their original packaging.</p>\n\t</li>\n\t<li>\n\t<p>Items purchased anywhere except arcteryx.com must be returned to the original place of purchase.</p>\n\t</li>\n</ul>',
      className: null,
    },
  ],
  right: [
    {
      content:
        '<h2>Can\'t Find what you are looking for?</h2>\n\n<p>If you are unable to find the information you are looking for in the customer support centre please contact one of our Customer Service Representatives and we will help answer your questions.</p>\n\n<p><strong>Customer Service Hours<!-- -->:</strong></p>\n\n<p data-pm-slice="1 1 []">Note: We are following recommendations from our local health authorities for COVID-19. Due to a high volume of contacts and limitations to our office capacity, our response may be delayed.</p>\n\n<p><strong><a class="replace-cs-phone" href="tel:1-866-458-2473">1-866-458-2473</a></strong><br />\nAmericas: 8am-5pm, Mon-Fri PST<br />\nEurope: 9am-5pm, Mon-Fri</p>\n\n<p><strong><a href="/help/contact">Contact Us Form</a></strong><br />\nCurrently, our response time is 2 to 3 business days.</p>\n',
      className: "help__icon-tooltip",
    },
    {
      content:
        '<h2 style="margin: 2em 0 1em;">PRODUCT SERVICE</h2>\n\n<p>Need help with your Arc\'teryx product?</p>\n\n<p style="margin: 1em 0 0;"><a href="/help/product-service">View our Product Service Page</a></p>\n',
      className: "help__icon-tooltip",
    },
  ],
};

const basicContent = {
  left: [
    {
      content:
        "Sit amet nulla facilisi morbi. Tortor pretium viverra suspendisse potenti nullam ac tortor vitae purus. Praesent tristique magna sit amet purus gravida quis. Sit amet justo donec enim diam vulputate ut pharetra sit. Et egestas quis ipsum suspendisse ultrices gravida dictum fusce ut. Nunc sed id semper risus in hendrerit gravida rutrum quisque. Libero justo laoreet sit amet cursus sit amet dictum sit. Lectus vestibulum mattis ullamcorper velit sed ullamcorper. Vel fringilla est ullamcorper eget nulla facilisi. Massa enim nec dui nunc mattis enim ut tellus elementum. Integer vitae justo eget magna fermentum iaculis. Eget nunc lobortis mattis aliquam faucibus purus in massa tempor. Mattis aliquam faucibus purus in massa tempor nec. Lectus vestibulum.",
    },
  ],
  right: [
    {
      content:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
    },
  ],
};

const helpSizingContent = {
  left: [
    {
      content:
        "Please note that garment fit and length may vary from season to season as designs are revised or materials are updated.",
      type: "alert",
      alertType: "info",
    },
    {
      content:
        "<p>Our garments are designed with fits appropriate to the intended activity that take into account ergonomics, comfort, layer systems and also to maximize the performance of fabric technologies.</p>",
    },
  ],
  right: [
    {
      content:
        '<h2>Can\'t Find what you are looking for?</h2>\n\n<p>If you are unable to find the information you are looking for in the customer support centre please contact one of our Customer Service Representatives and we will help answer your questions.</p>\n\n<p>Customer Service Hours:</p>\n\n<p><a class="replace-cs-phone" href="tel:1-866-458-2473">1-866-458-2473</a><br /><br />\nAmericas: 8am-5pm PST, Monday to Friday  <br />  <br />\nEurope: 9am-5pm, Monday to Friday</p>\n\n<p><a href="/help/contact">Contact Us Form</a><br />\n',
      className: "help__icon-size-tooltip",
    },
  ],
};

// Decorator
const storyWrapper = storyFn => <StoryWrapper>{storyFn()}</StoryWrapper>;

// Story
export default {
  title: "Layout Componet",
  parameters: {
    layout: "fullscreen",
  },
  decorators: [storyWrapper],
};

// Templates
const HelpBottomLayoutTemplate = props => <HelpBottomLayout {...props}>Hi </HelpBottomLayout>;

const OrderHistoryHelpBottomLayoutTemplate = props => (
  <OrderHistoryWrapper>
    <HelpBottomLayout {...props}>Hi </HelpBottomLayout>
  </OrderHistoryWrapper>
);

// Stories
export const SimpleHelpBottomLayout = HelpBottomLayoutTemplate.bind({});
SimpleHelpBottomLayout.args = {
  left: basicContent.left,
  right: basicContent.right,
};
SimpleHelpBottomLayout.storyName = "Generic Help Bottom Layout";

export const OtderHistoryHelpBottomLayout = OrderHistoryHelpBottomLayoutTemplate.bind({});
OtderHistoryHelpBottomLayout.args = {
  isHtml: true,
  hideTopBorder: true,
  left: cmscontent.left,
  right: cmscontent.right,
};
OtderHistoryHelpBottomLayout.storyName = "Order History Bottom Layout";

export const HelpSizingBottomLayout = HelpBottomLayoutTemplate.bind({});
HelpSizingBottomLayout.args = {
  isHtml: true,
  rightWidth: "24",
  leftWidth: "100",
  hideTopBorder: true,
  isHelpPageSizing: true,
  left: helpSizingContent.left,
  right: helpSizingContent.right,
};
HelpSizingBottomLayout.storyName = "Generic Help Sizing Layout";
