// @arcteryx/design-system/src/components/organisms/table/Table.stories.tsx
import React from "react";
import { type Meta, type StoryObj } from "@storybook/react";

import Table, { type TableProps } from "./Table";
import {
  cmTable,
  inchesTable,
  internationalTable,
} from "../../molecules/tableSizing/__mocks__/tableSizingMocks";

const meta: Meta = {
  title: "Design System/Organisms/Table",
  component: Table,
};

export default meta;

type Story = StoryObj<any>;

export const MensJacketNewSizing: Story = {
  render: (args: TableProps) => <Table {...args} />,
  args: {
    title: "Men's",
    disclaimer:
      "NOTE: Men's fit has been updated with a revised fit. Medium size remains unchanged. Smaller sizes are now tighter and larger sizes are now looser. If your measurements are between sizes, select a larger size for a looser fit or a smaller size for a tighter fit.",
    tables: [cmTable, inchesTable, internationalTable],
    category: "Jacket/Shirt Sizing Chart - New Sizing",
  },
};

export const MensJacketFormerSizing: Story = {
  render: (args: TableProps) => <Table {...args} />,
  args: {
    title: "Men's",
    disclaimer: "",
    tables: [cmTable, inchesTable, internationalTable],
    category: "Jacket/Shirt Sizing Chart - New Sizing",
  },
};
