// @arcteryx/components-content-blocks/src/components/PageLayout.jsx
/* istanbul ignore file */

import React from "react";
import { CtaContentBlock } from "./CtaContentBlock/CtaContentBlock";
import HeroSection from "./sections/HeroSection";
import Hero6Section from "./sections/Hero6Section";
import SecondarySection from "./sections/SecondarySection";
import TertiarySection from "./sections/TertiarySection";
import styled from "styled-components";

const sectionMap = {
  hero6: Hero6Section,
  hero: HeroSection,
  secondary: SecondarySection,
  tertiary: TertiarySection,
};

const StyledDiv = styled.div`
  display: grid;
  width: 100%;
  grid-row-gap: 1rem;
  background-color: ${({ backgroundColor = "white" }) => backgroundColor};
  z-index: 0;
  position: relative;
`;

export const PageLayout = ({ sections = [], backgroundColor = "white" }) => {
  return (
    <StyledDiv backgroundColor={backgroundColor}>
      {sections.map(({ layout, blocks = [] }, index) => {
        const SectionComponent = sectionMap[layout];
        return SectionComponent ? (
          <React.Fragment key={layout + index}>
            <SectionComponent data-testid={layout}>
              {blocks.map((block, blockIndex) => (
                <CtaContentBlock type={layout} key={blockIndex} {...block} />
              ))}
            </SectionComponent>
          </React.Fragment>
        ) : null;
      })}
    </StyledDiv>
  );
};
