// @arcteryx/design-system/src/components/molecules/card/__mocks__/cardMocks.ts
import { type CardProps } from "../Card.types";
import {
  fitGuideBlocks,
  measuremenBlocks,
  mensSizingBlocks,
  shoeBlocks,
} from "./cardRichText";

export const mensSizingChartCardData: CardProps = {
  variant: "product",
  button: {
    context: "Standard",
    onClick: () => null,
    label: "VIEW CHART",
    url: "https://arcteryx.com/ca/en/help/product-service?intcmp=product-service",
  },
  shouldOpenModal: true,
  content: mensSizingBlocks,
  image: {
    src: "https://cdn.sanity.io/images/inkbj32c/development/b20344aa741e8d44649030e3dabc3ef6f240b7f7-506x1800.jpg",
    alt: "card image",
  },
};

export const shoeSizingChartCardData: CardProps = {
  variant: "product",
  button: {
    context: "Standard",
    onClick: () => null,
    label: "VIEW CHART",
    url: "https://arcteryx.com/ca/en/help/product-service?intcmp=product-service",
  },
  content: shoeBlocks,
  image: {
    src: "https://images.arcteryx.com/sizingCharts/Aerios-FL-MID-GTX-Shoe-Black-Cinder.jpg",
    alt: "card image",
  },
};

export const imageCardData: CardProps = {
  variant: "image",
  content: [
    {
      _key: "h",
      _type: "block",
      style: "p",
      children: [
        {
          _key: "h1",
          _type: "span",
          text: "STRAIGHT HEM",
        },
      ],
    },
  ],

  image: {
    src: "https://images.arcteryx.com/sizingCharts/hems-s22/arcteryx_fitguide_mens_jacket_hem_straight_drop_01_comp_v02.png",
    alt: "card image",
  },
};

export const dynamicCardData: CardProps = {
  ...mensSizingChartCardData,
  variant: "dynamic",
  content: measuremenBlocks,
  button: undefined,
  image: {
    src: "https://cdn.sanity.io/images/inkbj32c/development/e8c94e936792182cff3b42ee5ff6625977edf091-336x420.png",
    alt: "card image",
  },
  title: "HOW TO FIND YOUR MEASUREMENTS:",
};

export const videoData = {
  videoSrc:
    "//images.arcteryx.com/sizingCharts/mens-tops/arcteryx_fitguide_mens_tops_fitted_01_conform_v01_H264.mp4",
  autoPlay: true,
  playsInline: true,
  loop: true,
  muted: true,
};

export const videoCardData: CardProps = {
  variant: "video",
  content: fitGuideBlocks,
  video: videoData,
};
