// @arcteryx/design-system/src/components/molecules/card/__mocks__/cardRichText.ts
import {
  type CustomPortableTextBlock,
  type RichTextBlocks,
} from "../../richText/RichText.types";

export const mensSizingBlocks: RichTextBlocks = [
  {
    _key: "2f6ed22ece05",
    _type: "block",
    children: [
      {
        _key: "230faea91e7c",
        _type: "span",
        marks: [],
        text: "MEN'S\nSIZING CHART",
      },
    ],
    markDefs: [],
    style: "normal",
  },
];

export const videoCardBlocks: RichTextBlocks = [
  {
    _type: "block",
    _key: "ec7f49c9ed48",
    style: "h3",
    markDefs: [],
    children: [
      {
        _type: "span",
        _key: "15db0e2b60c1",
        text: "FITTED",
        marks: [],
      },
    ],
  },
  {
    _key: "24964cd9eb3e",
    children: [
      {
        _type: "span",
        marks: [],
        text: "Our classic fit is cut comfortably throughout the chest, waist, hip, and thigh. It allows freedom of movement, provides shape, and layers comfortably under our regular and relaxed fit shells.",
        _key: "f4b6b69c21f90",
      },
    ],
    markDefs: [],
    _type: "block",
    style: "normal",
  },
];

export const measuremenBlocks: RichTextBlocks = [
  {
    _type: "block",
    _key: "08cae585827e",
    style: "normal",
    markDefs: [],
    children: [
      {
        text: "Note:",
        _type: "span",
        _key: "4aaee2651ba7",
        marks: ["strong"],
      },
      {
        _type: "span",
        _key: "bb7a737cd4ac",
        marks: [],
        text: " Calculate your own body measurements, then select your product size. ",
      },
    ],
  },
  {
    _type: "block",
    _key: "32d7cb27380e",
    style: "normal",
    markDefs: [],
    children: [
      {
        _type: "span",
        _key: "d99076a7f120",
        marks: ["strong"],
        text: "Sleeve",
      },
      {
        _type: "span",
        _key: "72ac177a9dad",
        marks: [],
        text: ": With arm relaxed at your side and slightly bent, measure from center back neck, across shoulders to elbow and down to wrist. ",
      },
    ],
    level: 1,
    listItem: "number",
  },
  {
    _type: "block",
    _key: "acd063bf930d",
    style: "normal",
    markDefs: [],
    level: 1,
    listItem: "number",
    children: [
      {
        _type: "span",
        _key: "48448e8fd46b",
        marks: ["strong"],
        text: "Chest",
      },
      {
        _type: "span",
        _key: "c5fc854e15bc",
        marks: [],
        text: " : With arms relazed at your sided, measure under your arms, around the fullest part of your chest. ",
      },
    ],
  },
  {
    _type: "block",
    _key: "f8f1303f4652",
    style: "normal",
    markDefs: [],
    level: 1,
    listItem: "number",
    children: [
      {
        _type: "span",
        _key: "ccb7e0f0aa60",
        marks: ["strong"],
        text: "Waist:",
      },
      {
        _type: "span",
        _key: "e74af4cd7f26",
        marks: [],
        text: " Measure Around your waist at the narrowest part. ",
      },
    ],
  },
  {
    _type: "block",
    _key: "46b7e3600eda",
    style: "normal",
    markDefs: [],
    level: 1,
    listItem: "number",
    children: [
      {
        _type: "span",
        _key: "f3b790cf154c",
        marks: ["strong"],
        text: "Hip:",
      },
      {
        _type: "span",
        _key: "6d6f473c252c",
        marks: [],
        text: " Standing with feet hip width apart, measure around your hops at the widest part. ",
      },
    ],
  },
];

const listItem: CustomPortableTextBlock = {
  _type: "block",
  _key: "46b7e3600eda",
  style: "normal",
  markDefs: [],
  level: 1,
  listItem: "number",
  children: [
    {
      _type: "span",
      _key: "6d6f473c252c",
      marks: [],
      text: " More...\n...\n...\n...",
    },
  ],
};

export const longMeasurementBlocks: RichTextBlocks = [
  ...measuremenBlocks,
  listItem,
  listItem,
  listItem,
  listItem,
  listItem,
];

export const shoeBlocks: CustomPortableTextBlock[] = [
  {
    _key: "2f6ed22ece05",
    _type: "block",
    children: [
      {
        _key: "230faea91e7c",
        _type: "span",
        marks: [],
        text: "FOOTWEAR\nSIZING CHART",
      },
    ],
    markDefs: [],
    style: "normal",
  },
];

export const fitGuideBlocks: RichTextBlocks = [
  {
    _key: "98cf20842003",
    _type: "block",
    children: [
      {
        _key: "4315230fe8150",
        _type: "span",
        marks: [],
        text: "FITTED",
      },
    ],
    markDefs: [],
    style: "h3",
  },
  {
    _key: "0422b74c55f2",
    _type: "block",
    children: [
      {
        _key: "299effd795f70",
        _type: "span",
        marks: [],
        text: "Our most generous fit is cut loosely throughout the entire body. Dropped shoulders and wide sleeves provide freedom of movement while offering drape and structure for a contemporary look.",
      },
    ],
    markDefs: [],
    style: "normal",
  },
];
