// @arcteryx/design-system/src/components/atoms/table-cell/TableCell.stories.tsx
import React from "react";
import TableCell from "./TableCell";
import styled from "styled-components";
import { type Meta, type StoryObj } from "@storybook/react";

const StyledContainer = styled.div`
  tbody {
    display: flex;
    flex-direction: column;
  }
  td {
    font-weight: 400;
    height: 2rem;
    line-height: 1.5;
    text-align: center;
  }
  td:first-of-type {
    font-weight: 600;
  }

  td:nth-child(even) {
    background-color: var(--colour-grey-background);
  }
  border-radius: 10px;
  border: 1px dashed #e5e5e5;
  display: flex;
  flex-direction: column;
  width: 150px;
`;

const initialProps: { row: string[] } = {
  row: ["Title", "10", "10", "12 %", "12 %", "23 ½", "25 ½"],
};

const meta: Meta<typeof TableCell> = {
  title: "Design System / Atoms / TableCell",
  component: TableCell,
};

export default meta;

type Story = StoryObj<any>;

export const Default: Story = {
  render: (args: { row: string[] }) => (
    <StyledContainer>
      <table>
        <tbody>
          {args.row.map((cell: string, index: number) => (
            <TableCell key={index} content={cell} />
          ))}
        </tbody>
      </table>
    </StyledContainer>
  ),
  args: initialProps,
};
