// @arcteryx/components-alerts-overlays/dist/cjs/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var styled = require('styled-components');
var PropTypes = require('prop-types');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);
var PropTypes__default = /*#__PURE__*/_interopDefaultLegacy(PropTypes);

const AlertStyles = styled__default["default"].div`
  font-family: var(--font-urw-din);
  font-size: 15px;
  line-height: 1.5em;
  min-height: 60px;
  display: flex;
  flex-direction: row;
  position: relative;
  justify-content: flex-start;
  align-items: flex-start;
  padding: 0 1rem;
  > span {
    margin-left: 27.5px;
    padding: 18px 0;
    dispaly: table-cell;
    vertical-align: center;
  }
  p {
    margin: 0 0 1.5rem 0 !important;
    max-width: none !important;

    &:last-of-type {
      margin-bottom: 0 !important;
    }
  }
  &:before {
    display: inline-block;
    width: 20px;
    height: 20px;
    position: absolute;
    top: 30px;
    transform: translateY(-50%);
  }
  ${_ref => {
  let {
    type
  } = _ref;
  if (type === "info") {
    return `
        background: var(--background-blue);
        border: 1px solid var(--focus-blue);
        color: var(--info-blue);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Information_Icon_Blue.svg);
        }
        p, a {
          color: var(--info-blue) !important;
        }
      `;
  }
  if (type === "caution") {
    return `
        background: var(--background-yellow);
        border: 1px solid var(--medium-yellow);
        color: var(--caution-yellow);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Caution_Icon.svg);
        }
        p, a {
          color: var(--caution-yellow) !important;
        }
      `;
  }
  if (type === "warning") {
    return `
        background: var(--background-red);
        border: 1px solid var(--medium-red);
        color: var(--warning-red);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Warning_Icon.svg);
        }
        p, a {
          color: var(--warning-red) !important;
        }
      `;
  }
  /* istanbul ignore else */
  if (type === "success") {
    return `
        background: var(--background-green);
        border: 1px solid var(--medium-green);
        color: var(--success-green);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Success_Icon.svg);
        }
        p, a {
          color: var(--success-green) !important;
        }
      `;
  }
  /* istanbul ignore else */
  if (type === "advisory") {
    return `
        background: var(--colour-grey-background);
        border: 1px solid  var(--warning-red);
        color: var(--warning-red);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Information_Icon_Red.svg);
        }
        p, a {
          color: var(--warning-red) !important;
        }
      `;
  }
}}
`;
const Alert = _ref2 => {
  let {
    type,
    dangerouslySetInnerHTML,
    children,
    className
  } = _ref2;
  if (type) {
    return /*#__PURE__*/React__default["default"].createElement(AlertStyles, {
      className: className,
      type: type
    }, dangerouslySetInnerHTML ? /*#__PURE__*/React__default["default"].createElement("span", {
      dangerouslySetInnerHTML: {
        __html: children
      }
    }) : /*#__PURE__*/React__default["default"].createElement("span", null, children));
  }
  return null;
};
Alert.defaultProps = {
  type: "info"
};
Alert.prototype = {
  type: PropTypes__default["default"].string.isRequired
};

const TOAST_TYPE = {
  SUCCESS: "success",
  WARNING: "warning",
  CAUTION: "caution"
};
const ToastAnimate = styled.keyframes`
  0% {
    opacity: 0.5;
    transform: translateY(13.3rem);
  }
  25% {
    opacity: 0.8;
    transform: translateY(-20rem);
  }
  50% {
    opacity: 1;
    transform: translateY(-20rem);
  }
  75% {
    opacity: 1;
    transform: translateY(-20rem);
  }
`;
const Wrapper = styled__default["default"].div`
  display: flex;
  justify-content: center;
  position: fixed;
  width: 100%;
  height: 100%;
  left:0;
`;
const ToastStyles = styled__default["default"].div`
  font-family: var(--font-urw-din);
  font-size: 1rem;
  line-height: 1.5em;
  min-height: 4rem;
  display: flex;
  flex-direction: row;
  position: fixed;
  justify-content: flex-start;
  align-items: flex-start;
  padding: 0 1rem;
  bottom: -13.3rem;
  span {
    margin-left: 1.83rem;
    padding: 1.125rem 0;
    dispaly: table-cell;
    vertical-align: center;
    max-width: 80ch;
  }
  p {
    margin: 0 0 1.5rem 0 !important;
    max-width: 80ch !important;

    &:last-of-type {
      margin-bottom: 0 !important;
    }
  }
  &:before {
    display: inline-block;
    width: 1.3rem;
    height: 1.3rem;
    position: absolute;
    top: 2rem;
    transform: translateY(-50%);
  }
  animation: ${ToastAnimate} 5s;
  ${_ref => {
  let {
    type
  } = _ref;
  if (type === TOAST_TYPE.CAUTION) {
    return `
        background: var(--background-yellow);
        border: 1px solid var(--medium-yellow);
        color: var(--caution-yellow);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Caution_Icon.svg);
        }
        p, a {
          color: var(--caution-yellow) !important;
        }
      `;
  }
  if (type === TOAST_TYPE.WARNING) {
    return `
        background: var(--background-red);
        border: 1px solid var(--medium-red);
        color: var(--warning-red);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Warning_Icon.svg);
        }
        p, a {
          color: var(--warning-red) !important;
        }
      `;
  }
  if (type === TOAST_TYPE.SUCCESS) {
    return `
        background: var(--background-green);
        border: 1px solid var(--medium-green);
        color: var(--success-green);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Success_Icon.svg);
        }
        p, a {
          color: var(--success-green) !important;
        }
      `;
  }
}}
`;
const Toast = _ref2 => {
  let {
    type,
    dangerouslySetInnerHTML,
    children,
    className
  } = _ref2;
  if (type) {
    return /*#__PURE__*/React__default["default"].createElement(Wrapper, null, /*#__PURE__*/React__default["default"].createElement(ToastStyles, {
      className: className,
      type: type
    }, dangerouslySetInnerHTML ? /*#__PURE__*/React__default["default"].createElement("span", {
      dangerouslySetInnerHTML: {
        __html: children
      }
    }) : /*#__PURE__*/React__default["default"].createElement("span", null, children)));
  }
  return null;
};
Toast.defaultProps = {
  type: TOAST_TYPE.CAUTION
};
Toast.prototype = {
  type: PropTypes__default["default"].string.isRequired
};

exports.Alert = Alert;
exports.Toast = Toast;
