// @arcteryx/components-layouts/src/components/Style.js
import styled from "styled-components";
import { Breakpoint } from "@arcteryx/components-typography";

const COLUMN_WIDTH_DEFAULT = "40%";

const getPadding = props => {
  if (props.isHelpPageSizing) {
    return "0 0 0 70px";
  } else if (props.noTopPadding) {
    return "0 0 1rem 2rem";
  } else {
    return "1em 0 1rem 2rem";
  }
};

export const AlertWrapper = styled.div`
  ${Breakpoint("xs")`
  margin: 0 0 3rem 0;
`}
  ${Breakpoint("md")`
margin: 0 3rem 3rem 0;
`}
`;

export const ContentWrapper = styled.div`
  width: 100%;
  margin: 0 auto 2rem;
  padding: ${props => (props.hideTopBorder ? "0" : "1em 0")};
  border-top: ${props => (props.hideTopBorder ? "none" : "1px solid var(--colour-grey-pinline)")};
`;
export const InnerWrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: ${props => (props.isHelpPageSizing ? "88%" : "100%")};
  max-width: 1430px;

  ${Breakpoint("md")`
flex-wrap: nowrap;
`}

  a {
    color: var(--colour-black);
    text-decoration: underline;
  }
`;

export const RightContent = styled.div`
  width: 100%;
  padding: ${props => (props.noTopPadding ? "0" : "1em 0 0 0")};
  margin: 1em 0;

  ${Breakpoint("md")`
    width: ${props => (props.rightWidth ? `${props.rightWidth}%` : COLUMN_WIDTH_DEFAULT)};
    padding: ${props => getPadding(props)};
    margin: ${props => (props.isHelpPageSizing ? "0" : "0 0 0 2rem")};
    border-left: 1px solid var(--colour-grey-pinline);
    position: relative;
  `}
`;

export const LeftContent = styled.div`
  padding: ${props => (props.noTopPadding ? "0" : "1em 0 0 0")};
  width: 100%;
  position: relative;
  flex: 1 1;

  ${Breakpoint("md")`
    width: ${props => (props.leftWidth ? `${props.leftWidth}%` : "60%")};
    padding: ${props => (props.noTopPadding ? "0" : "1em 0")};
  `}
`;
export const RightInnerContent = styled.div`
  position: relative;

  &.help__icon-size-tooltip {
    ${Breakpoint("xs")`
      padding: ${({ isHelpPageSizing }) => (isHelpPageSizing ? "0 1rem 0 3rem" : "3rem")};
    `}

    ${Breakpoint("md")`
      padding: ${({ isHelpPageSizing }) => (isHelpPageSizing ? "0" : "0 3rem")};
    `}
  }

  &.help__icon-size-tooltip:after {
    content: url(//images.arcteryx.com/foundation-ui/svgs/Question_Icon.svg);
    position: absolute;
    left: 0;
    top: 0;
    width: 30px;
    height: 30px;
    display: block;

    ${Breakpoint("xs")`
      margin-left: 0;
    `}

    ${Breakpoint("md")`
      margin-left: ${({ isHelpPageSizing }) => (isHelpPageSizing ? "-3rem" : "0")};
    `}
  }
`;
