// @arcteryx/design-system/src/components/molecules/linkBox/LinkBox.tsx
import React from "react";
import styled from "styled-components";
import { Breakpoint } from "@arcteryx/components-typography";
import { Icon } from "@arcteryx/components-icons";
import { type LinkBoxProps } from "./LinkBox.types";

const StyledLink = styled.a`
  border: 1px solid var(--colour-grey-pinline);
  display: flex;
  width: 100%;
  height: auto;
  min-height: 100px;
  justify-content: center;
  vertical-align: middle;
  text-decoration: none;
  &:hover {
    text-decoration: underline;
    color: var(--colour-black);
  }
`;
const StyledWrapper = styled.span`
  display: flex;
  align-items: center;
  cursor: pointer;
  flex-direction: column;
  gap: 1rem;
  padding: 1rem;
  min-height: 110px;

  ${Breakpoint("sm")`
  flex-direction: row;
  min-height: auto;
`}
`;

const StyledText = styled.span`
  flex: 1;
  font-size: 15px;
  line-height: 15px;
  font-weight: 700;
  font-family: var(--font-urw);
  text-transform: uppercase;
  text-align: center;
  color: #000;
  &:hover {
    text-decoration: underline;
    color: var(--colour-black);
  }
`;

export const LinkBox: React.FC<LinkBoxProps> = (props) => {
  const { className, icon, label, url, target } = props;

  return (
    <StyledLink
      target={target}
      className={className}
      href={url}
      aria-label={label}
    >
      <StyledWrapper>
        <Icon type={icon} aria-hidden="true" width="30" />
        <StyledText>{label}</StyledText>
      </StyledWrapper>
    </StyledLink>
  );
};
