// @arcteryx/design-system/src/components/molecules/linkBox/LinkBox.stories.tsx
import { type Meta } from "@storybook/react";
import React from "react";
import { LinkBox } from "./LinkBox";
import styled from "styled-components";
import { Breakpoint } from "@arcteryx/components-typography";
import { type LinkBoxProps } from "./LinkBox.types";

// This is a styled component that is used to display multiple LinkBoxes in a grid, we use this to test the responsiveness of the component
const StyledContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 1rem;
  ${Breakpoint("lg")`
  grid-gap: 2rem;
  border: 1px solid var(--colour-grey-medium);
    margin: 3rem;
    padding: 3rem;
`}
`;

const meta: Meta<typeof LinkBox> = {
  component: LinkBox,
  title: "Design System/Molecules/LinkBox",
  argTypes: {},
};
export default meta;

interface StoryProps {
  boxes: LinkBoxProps[];
}

const MultipleLinkBoxesArgs: StoryProps = {
  boxes: [
    {
      url: "//arcteryx.com/help/sizing?intcmp=sizing",
      icon: "play-button",
      label: "Sizing & Fit Guide",
      target: "_blank",
    },
    {
      url: "//example.com/another-link",
      icon: "star",
      label: "Product Service",
    },
    {
      url: "//example.com/more-link",
      icon: "left-arrow",
      label: "Product Care",
    },
    {
      url: "//example.com/more-link",
      icon: "down-arrow",
      label: "Returns & refunds",
    },
  ],
};

const MultipleLinkBoxesWrapper: React.FC<StoryProps> = ({
  boxes,
}): JSX.Element => {
  return (
    <StyledContainer>
      {boxes.map((box) => (
        <LinkBox key={box.url} {...box} />
      ))}
    </StyledContainer>
  );
};

export const MultipleLinkBoxes = (args: StoryProps): JSX.Element => (
  <MultipleLinkBoxesWrapper {...args} />
);

MultipleLinkBoxes.args = { ...MultipleLinkBoxesArgs };
