// @arcteryx/design-system/src/components/organisms/boxListImage/__mocks__/boxListImageMocks.tsx
import React from "react";
import { type ImageProps } from "../../../atoms";
import { RichText, type RichTextProps } from "../../../molecules/richText";
import {
  RichTextIcon,
  type RichTextIconProps,
} from "../../../molecules/richTextIcon";
import { type BoxListImageProps } from "../BoxListImage";

export const richtextBlocks: RichTextProps = {
  content: [
    {
      _type: "block",
      style: "h2",
      children: [
        {
          text: "Design Your Purpose. Live Our Values.",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
    {
      _type: "block",
      style: "normal",
      children: [
        {
          text: "Our values are our DNA - embedded in what we do, and how we do it. When you join our team, it's not just a one-way deal of you bringing your remarkable talent to our collective. These four statements define how we show up in the world and are also our promises to you.",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
  ],
};

export const imageBlock: ImageProps = {
  alt: "Climbers scale a rock face in the Pacific Northwest",
  src: "https://cdn.sanity.io/images/inkbj32c/development/882ba3acb8bcebf710bdb01704e5125d2acc5ba6-551x736.png?q=75",
};

export const richTextIconTree: RichTextIconProps = {
  icon: "tree",
  content: [
    {
      _type: "block",
      style: "h3",
      children: [
        {
          text: "DISRUPTIVE EVOLUTION",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
    {
      _type: "block",
      style: "normal",
      children: [
        {
          text: "Design your future through challenge, experimentation, and inspiration.",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
    {
      _type: "block",
      style: "normal",
      children: [
        {
          text: "You create your own life and future. We support you to find opportunities to grow and innovate, challenge the status quo, and seek a better way as a collective.",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
  ],
};

export const richTextIconMountain: RichTextIconProps = {
  icon: "mountain",
  content: [
    {
      _type: "block",
      style: "h3",
      children: [
        {
          text: "COMMIT",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
    {
      _type: "block",
      style: "normal",
      children: [
        {
          text:
            "Set and deliver on bold objectives as we collectively raise the bar. \n" +
            "\n" +
            "We empower you to live in your strengths, commit to what lights you up, and unleash your talent to advance yourself, your team, and the brand. Together, we design what is possible.",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
  ],
};

export const richTextIconSun: RichTextIconProps = {
  icon: "sun",
  content: [
    {
      _type: "block",
      style: "h3",
      children: [
        {
          text: "LIVE IT",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
    {
      _type: "block",
      style: "normal",
      children: [
        {
          text:
            "Deepen your connection to nature and live your most purposeful life.\n" +
            "\n" +
            "The transformative power of nature is profound. We believe in taking care of your whole self and that work-life integration requires restorative outlets. Explore, and do, what is most meaningful to you.",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
  ],
};

export const richTextIconGlobe2: RichTextIconProps = {
  icon: "globe-2",
  content: [
    {
      _type: "block",
      style: "h3",
      children: [
        {
          text: "LEAVE IT BETTER",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
    {
      _type: "block",
      style: "normal",
      children: [
        {
          text:
            "Create a better world, together.\n" +
            "\n" +
            "What matters to you matters to us. We hold each other accountable to do the right thing, not the easy thing, for our people, our brand, our communities, and environment.",
          _type: "span",
          marks: [],
        },
      ],
      markDefs: [],
    },
  ],
};

const RichTextIconTreeComponent: React.FC = () => (
  <RichTextIcon
    icon={richTextIconTree.icon}
    content={richTextIconTree.content}
  />
);

const RichTextIconMountainComponent: React.FC = () => (
  <RichTextIcon
    icon={richTextIconMountain.icon}
    content={richTextIconMountain.content}
  />
);

const RichTextIconSunComponent: React.FC = () => (
  <RichTextIcon icon={richTextIconSun.icon} content={richTextIconSun.content} />
);

const RichTextIconGlobe2Component: React.FC = () => (
  <RichTextIcon
    icon={richTextIconGlobe2.icon}
    content={richTextIconGlobe2.content}
  />
);

const RichTextComponent: React.FC = () => (
  <RichText content={richtextBlocks.content} textAlign="left" />
);

export const boxListImageStoryMock: BoxListImageProps = {
  displayImageBelowMd: true,
  image: imageBlock,
  children: [
    <RichTextComponent key="1" />,
    <RichTextIconTreeComponent key="2" />,
    <RichTextIconMountainComponent key="3" />,
    <RichTextIconSunComponent key="4" />,
    <RichTextIconGlobe2Component key="5" />,
  ],
};
