// @arcteryx/design-system/src/components/organisms/Pagination/__mocks__/jobMockData.ts
interface Job {
  id: string;
  title: string;
  team: string;
  location: string;
  department: string;
  commitment: string;
  url: string;
}

export const jobMockData: Job[] = [
  {
    id: "dac063ab-2738-4a24-93cd-7f18ad2cb9c7",
    title: "Allocations Coordinator - Bilingual",
    team: "Merchandise Planning & Allocations",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/dac063ab-2738-4a24-93cd-7f18ad2cb9c7",
  },
  {
    id: "3c3f8f0a-1431-4cb6-8d7c-56a806e91edf",
    title: "Business Analyst, Data & Analytics",
    team: "Data & Analytics",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/3c3f8f0a-1431-4cb6-8d7c-56a806e91edf",
  },
  {
    id: "10404f35-0784-4b56-ae28-77dfe948eb4d",
    title: "Community Marketing Manager – Germany",
    team: "EMEA",
    location: "Garching, Munich",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/10404f35-0784-4b56-ae28-77dfe948eb4d",
  },
  {
    id: "7c1bf605-2ced-4a5a-b293-c0555d22f752",
    title: "Construction Project Manager",
    team: "Store Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/7c1bf605-2ced-4a5a-b293-c0555d22f752",
  },
  {
    id: "84169b1e-0ef4-4ded-921e-84e20da8662f",
    title: "Directeur régional – Paris | Regional Manager - Paris",
    team: "EMEA",
    location: "Paris, France",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/84169b1e-0ef4-4ded-921e-84e20da8662f",
  },
  {
    id: "2179de02-6e3a-4710-9378-01afe255075a",
    title: "Director, Design – Tokyo Design Center",
    team: "Tokyo Design Centre",
    location: "Tokyo",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/2179de02-6e3a-4710-9378-01afe255075a",
  },
  {
    id: "ebe82030-fdd5-4457-83e7-8b09e907515a",
    title: "Director, Global Footwear Product",
    team: "Footwear",
    location: "Portland, OR",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/ebe82030-fdd5-4457-83e7-8b09e907515a",
  },
  {
    id: "7d8b93b7-cb60-4f36-8527-b189c8f41791",
    title: "Director, Merchandising - Footwear",
    team: "Merchandising",
    location: "Portland, OR",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/7d8b93b7-cb60-4f36-8527-b189c8f41791",
  },
  {
    id: "0f03dc74-419d-4929-a02e-95cafe82b622",
    title: "Director, People & Culture Business Partnerships - Supply Chain",
    team: "Business Partnership",
    location: "North Vancouver, BC (Corporate)",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/0f03dc74-419d-4929-a02e-95cafe82b622",
  },
  {
    id: "74042140-a865-4e38-9da2-b5af6fa2542c",
    title: "Director, Retail Stores - Canada",
    team: "North America Commercial",
    location: "Vancouver, BC",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/74042140-a865-4e38-9da2-b5af6fa2542c",
  },
  {
    id: "19d259c9-d981-4d58-b0da-ec8743c3a653",
    title: "Director, Retail Stores - US East",
    team: "North America Commercial",
    location: "New York, NY",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/19d259c9-d981-4d58-b0da-ec8743c3a653",
  },
  {
    id: "cbe6c499-ff3c-458f-bcf0-fd2ba9f26c97",
    title: "Director, Retail Stores - US West",
    team: "North America Commercial",
    location: "Seattle, WA",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/cbe6c499-ff3c-458f-bcf0-fd2ba9f26c97",
  },
  {
    id: "b8c13e5f-f0c9-415c-9537-e6e2131f866c",
    title: "Executive Assistant",
    team: "Footwear",
    location: "Portland, OR",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/b8c13e5f-f0c9-415c-9537-e6e2131f866c",
  },
  {
    id: "8e95691f-c1f5-4e0c-8956-41a7e8f18e47",
    title: "Executive Assistant - EMEA",
    team: "EMEA",
    location: "London, England",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/8e95691f-c1f5-4e0c-8956-41a7e8f18e47",
  },
  {
    id: "84c3e844-00db-4897-86a3-e0b2648f0e31",
    title: "Female Fitting Model - Size 4-Small - Casual",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/84c3e844-00db-4897-86a3-e0b2648f0e31",
  },
  {
    id: "64c5c8ca-c905-4741-b069-cd0ce9ad0111",
    title: "Forecast Analyst - EMEA",
    team: "EMEA",
    location: "Garching, Munich",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/64c5c8ca-c905-4741-b069-cd0ce9ad0111",
  },
  {
    id: "e2a0dc4d-70cc-4dd0-9dcb-af5be840d14b",
    title: "Guest Services Representative - Temporary",
    team: "Guest Services",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "FT Temp/Seasonal",
    url: "https://jobs.lever.co/arcteryx.com/e2a0dc4d-70cc-4dd0-9dcb-af5be840d14b",
  },
  {
    id: "1bf58f91-24c5-417b-bae4-7be84be69157",
    title: "Guest Services Representative – Bilingual – Temporary",
    team: "Guest Services",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "FT Temp/Seasonal",
    url: "https://jobs.lever.co/arcteryx.com/1bf58f91-24c5-417b-bae4-7be84be69157",
  },
  {
    id: "0e897e35-4927-44c2-9ebe-1f2beeae72cf",
    title: "Internal Control and Risk Management Manager",
    team: "Finance",
    location: "North Vancouver, BC (Corporate)",
    department: "Finance",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/0e897e35-4927-44c2-9ebe-1f2beeae72cf",
  },
  {
    id: "704af038-1a01-4b3a-9fb0-98bbb72b1605",
    title: "Male Fitting Model - Size Large - Casual",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Contractor",
    url: "https://jobs.lever.co/arcteryx.com/704af038-1a01-4b3a-9fb0-98bbb72b1605",
  },
  {
    id: "de3d3566-a8ec-4370-90f9-c28505ec9e00",
    title: "Male Fitting Model - Size Medium - Casual",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Contractor",
    url: "https://jobs.lever.co/arcteryx.com/de3d3566-a8ec-4370-90f9-c28505ec9e00",
  },
  {
    id: "8a148005-a446-4022-8a40-57b24e1d13f8",
    title: "Manager, Delivery and Transportation",
    team: "Warehousing & Logistics",
    location: "New Westminster, BC",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/8a148005-a446-4022-8a40-57b24e1d13f8",
  },
  {
    id: "803e03d2-85c5-41c3-b21d-ba6a2a3b02d7",
    title: "Manager, Inventory Accounting",
    team: "Finance",
    location: "North Vancouver, BC (Corporate)",
    department: "Finance",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/803e03d2-85c5-41c3-b21d-ba6a2a3b02d7",
  },
  {
    id: "ec3e4566-916c-4bd4-88ec-a71b73171345",
    title: "Manager, Merchandising - EMEA",
    team: "EMEA",
    location: "Garching, Munich",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/ec3e4566-916c-4bd4-88ec-a71b73171345",
  },
  {
    id: "372b9c25-6c53-40e5-88a9-3b3b995b8f65",
    title: "Manager, Merchandising – Hardgoods & Accessories",
    team: "Merchandising",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/372b9c25-6c53-40e5-88a9-3b3b995b8f65",
  },
  {
    id: "148ba778-eead-4cf1-b901-ea7d3de5dc2a",
    title: "Manager, Raw Materials Sourcing",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/148ba778-eead-4cf1-b901-ea7d3de5dc2a",
  },
  {
    id: "3b57a6ba-6be0-43af-9a07-b1c06af928eb",
    title: "Materials Developer 1, Trims",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/3b57a6ba-6be0-43af-9a07-b1c06af928eb",
  },
  {
    id: "b40def81-af93-42e7-b06a-e9493a8cfa24",
    title: "Materials Developer 2 - Temporary",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "FT Temp/Seasonal",
    url: "https://jobs.lever.co/arcteryx.com/b40def81-af93-42e7-b06a-e9493a8cfa24",
  },
  {
    id: "4b037e4f-f82a-452f-ac45-255e4f14ab07",
    title: "Materials Developer 2, Hardgoods",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/4b037e4f-f82a-452f-ac45-255e4f14ab07",
  },
  {
    id: "e6b07337-fc8d-4ae7-a34d-9a5e2d13a4eb",
    title: "Merchandise Planner - A",
    team: "Merchandise Planning & Allocations",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/e6b07337-fc8d-4ae7-a34d-9a5e2d13a4eb",
  },
  {
    id: "3dd83433-4285-43da-bab3-29747b857209",
    title: "Merchandise Planner - B",
    team: "Merchandise Planning & Allocations",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/3dd83433-4285-43da-bab3-29747b857209",
  },
  {
    id: "77b0ea7f-75d8-45c3-aee3-80cca6e5b3ed",
    title: "Omnichannel Project Specialist",
    team: "Retail Strategic Initiatives",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/77b0ea7f-75d8-45c3-aee3-80cca6e5b3ed",
  },
  {
    id: "03f74d47-7405-443f-874c-d82c1ec23251",
    title: "Omnichannel Project Specialist - Temporary",
    team: "Retail Strategic Initiatives",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "FT Temp/Seasonal",
    url: "https://jobs.lever.co/arcteryx.com/03f74d47-7405-443f-874c-d82c1ec23251",
  },
  {
    id: "de520532-7b2c-46b0-ae48-443efb6caecd",
    title: "Pattern Maker/Developer 2",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/de520532-7b2c-46b0-ae48-443efb6caecd",
  },
  {
    id: "c17b1655-be45-45bc-9acf-256f600d64ed",
    title: "Payroll Administrator",
    team: "Finance",
    location: "North Vancouver, BC (Corporate)",
    department: "Finance",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c17b1655-be45-45bc-9acf-256f600d64ed",
  },
  {
    id: "e4228da0-da53-4524-a3c1-f4c56aed8f67",
    title: "People & Culture Business Partner – Corporate Centers",
    team: "Business Partnership",
    location: "North Vancouver, BC (Corporate)",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/e4228da0-da53-4524-a3c1-f4c56aed8f67",
  },
  {
    id: "7b3bf6e6-142b-49fd-8b02-65f7b249ac5b",
    title: "People & Culture Generalist - Manufacturing",
    team: "Business Partnership",
    location: "New Westminster, BC",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/7b3bf6e6-142b-49fd-8b02-65f7b249ac5b",
  },
  {
    id: "c7fc2a2c-aeeb-44b4-906a-efdaa2155a86",
    title: "Product Developer 2 - Hardgoods",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c7fc2a2c-aeeb-44b4-906a-efdaa2155a86",
  },
  {
    id: "25160d91-b984-4f18-8572-35a1a88002c7",
    title: "Product Operations Manager - Footwear",
    team: "Product Operations",
    location: "Portland, OR",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/25160d91-b984-4f18-8572-35a1a88002c7",
  },
  {
    id: "83e443e4-4cad-4196-bbaf-15c72107624e",
    title: "Product Owner, Omnichannel POS",
    team: "Retail Strategic Initiatives",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/83e443e4-4cad-4196-bbaf-15c72107624e",
  },
  {
    id: "657453cb-3f00-4c7e-90d8-b5f4fd49348c",
    title: "Quality Specialist - Danang, Vietnam",
    team: "Sourcing",
    location: "Danang",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/657453cb-3f00-4c7e-90d8-b5f4fd49348c",
  },
  {
    id: "5b742ddf-a0f0-4b93-84db-97cc94c1d623",
    title: "Quality Specialist - Hambantota, Sri Lanka",
    team: "Sourcing",
    location: "Hambantota",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/5b742ddf-a0f0-4b93-84db-97cc94c1d623",
  },
  {
    id: "d97234fc-c6d6-4275-ab93-8f6c55d90dd1",
    title: "Quality Specialist - Jakarta, Indonesia",
    team: "Sourcing",
    location: "Jakarta",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/d97234fc-c6d6-4275-ab93-8f6c55d90dd1",
  },
  {
    id: "e04ba67a-1b93-469a-9f2c-fe3c792e17c7",
    title: "Quality Specialist - Shenzhen, China",
    team: "Sourcing",
    location: "Shenzhen",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/e04ba67a-1b93-469a-9f2c-fe3c792e17c7",
  },
  {
    id: "c4fd5fff-0bca-431f-bf7f-dadefffa1dfc",
    title: "Quality Specialist - Vietnam North",
    team: "Sourcing",
    location: "Vietnam North",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c4fd5fff-0bca-431f-bf7f-dadefffa1dfc",
  },
  {
    id: "9fcacf06-8f56-4341-a1d5-fd863cf56cca",
    title: "Raw Materials Sourcing Specialist",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/9fcacf06-8f56-4341-a1d5-fd863cf56cca",
  },
  {
    id: "0223d917-8273-4459-87a0-29d2c63496ac",
    title: "Regional Manager, Central - EMEA",
    team: "EMEA",
    location: "Garching, Munich",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/0223d917-8273-4459-87a0-29d2c63496ac",
  },
  {
    id: "6844fe12-d5df-4d3f-980f-452fe8488872",
    title: "Regional Manager, NYC",
    team: "North America Commercial",
    location: "New York, NY",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/6844fe12-d5df-4d3f-980f-452fe8488872",
  },
  {
    id: "5ab3affc-884d-4237-aad7-082d0bb436d0",
    title: "Regional Manager, Tri-State",
    team: "North America Commercial",
    location: "USA East Coast",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/5ab3affc-884d-4237-aad7-082d0bb436d0",
  },
  {
    id: "5740c50e-85c5-4bc0-9954-70b1b331119f",
    title: "Senior Business Analyst, ERP (SAP)",
    team: "ERP & Business Applications",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/5740c50e-85c5-4bc0-9954-70b1b331119f",
  },
  {
    id: "1f85ca49-e598-46da-8c6f-e07849a67019",
    title: "Senior Compensation Analyst – Retail",
    team: "Total Rewards & People Operations",
    location: "North Vancouver, BC (Corporate)",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/1f85ca49-e598-46da-8c6f-e07849a67019",
  },
  {
    id: "ee43b7bc-745c-4768-a7d7-d077fc6ad8d6",
    title: "Senior Design Developer – Men’s Apparel",
    team: "Product Design",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/ee43b7bc-745c-4768-a7d7-d077fc6ad8d6",
  },
  {
    id: "6ee78b1e-711a-42b3-8aa7-2adfc0fa0fb6",
    title: "Senior Designer - Footwear",
    team: "Footwear",
    location: "Portland, OR",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/6ee78b1e-711a-42b3-8aa7-2adfc0fa0fb6",
  },
  {
    id: "68879563-ffdd-42d5-95e8-53d00335a192",
    title: "Senior Manager, People & Culture Business Partner - Temporary",
    team: "Business Partnership",
    location: "North Vancouver, BC (Corporate)",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/68879563-ffdd-42d5-95e8-53d00335a192",
  },
  {
    id: "410e0ea0-d706-4e50-b7f6-30e54cb11706",
    title: "Senior Manager, People & Culture Business Partnerships – Arc’One",
    team: "Business Partnership",
    location: "New Westminster, BC",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/410e0ea0-d706-4e50-b7f6-30e54cb11706",
  },
  {
    id: "1c84f4d8-a4b4-411f-9ee3-0092f72d4020",
    title: "Senior Material Quality Specialist, Footwear",
    team: "Sourcing",
    location: "Dongguan",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/1c84f4d8-a4b4-411f-9ee3-0092f72d4020",
  },
  {
    id: "4d0eea0e-1c1d-4aa6-a77d-950807f48420",
    title: "Senior Materials Developer",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/4d0eea0e-1c1d-4aa6-a77d-950807f48420",
  },
  {
    id: "dc0b3ddf-da31-469a-88a6-2b06a6b22605",
    title: "Senior Pattern Maker",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/dc0b3ddf-da31-469a-88a6-2b06a6b22605",
  },
  {
    id: "d8a84433-8525-41f9-b6c7-e1b8d5cf0914",
    title: "Senior Project Manager, IT PMO",
    team: "Cyber Security",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/d8a84433-8525-41f9-b6c7-e1b8d5cf0914",
  },
  {
    id: "b2875df8-b553-486a-a233-e03b5e835ba8",
    title: "Senior SAP Project Manager",
    team: "ERP & Business Applications",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/b2875df8-b553-486a-a233-e03b5e835ba8",
  },
  {
    id: "59bbe9e3-b744-490e-8ddb-233a7beef788",
    title: "Senior Sourcing Specialist, Raw Materials",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/59bbe9e3-b744-490e-8ddb-233a7beef788",
  },
  {
    id: "0bfbc194-c1aa-4c31-a855-40d82ec56b88",
    title: "Senior Technical Developer, Tokyo Creation Center",
    team: "Product Creation Development",
    location: "Tokyo",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/0bfbc194-c1aa-4c31-a855-40d82ec56b88",
  },
  {
    id: "44326403-ce2e-4859-9a84-e1a405500b12",
    title: "Software Quality Analyst",
    team: "IS Operations",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/44326403-ce2e-4859-9a84-e1a405500b12",
  },
  {
    id: "8cd16968-8418-4a85-9150-fdaacbd998f6",
    title: "Sourcing Manager - Hong Kong",
    team: "Sourcing",
    location: "Hong Kong",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/8cd16968-8418-4a85-9150-fdaacbd998f6",
  },
  {
    id: "90500af5-13c3-4d3f-959a-550f817cd068",
    title: "Studio Manager, Tokyo Creation Center",
    team: "Tokyo Design Centre",
    location: "Tokyo",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/90500af5-13c3-4d3f-959a-550f817cd068",
  },
  {
    id: "c6ce5080-c864-45f7-ad64-2c75f406b6e0",
    title: "Supply Planner",
    team: "Sourcing",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c6ce5080-c864-45f7-ad64-2c75f406b6e0",
  },
  {
    id: "05687487-b72c-4635-92c9-d50d58b4d365",
    title: "Systems & Process Specialist – Global Planning",
    team: "Product Operations",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/05687487-b72c-4635-92c9-d50d58b4d365",
  },
  {
    id: "03dafc0e-de12-41e5-bb1a-b7d37df21070",
    title: "Technical Architect",
    team: "Architecture",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/03dafc0e-de12-41e5-bb1a-b7d37df21070",
  },
  {
    id: "2afef199-c1a8-45b2-b2ae-beb481152e57",
    title: "Technical Developer 2",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/2afef199-c1a8-45b2-b2ae-beb481152e57",
  },
  {
    id: "03ac6831-844b-4d41-b965-4662aac913dd",
    title: "Technical Specialist Level 1 - China",
    team: "Sourcing",
    location: "China North",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/03ac6831-844b-4d41-b965-4662aac913dd",
  },
  {
    id: "89d4cb82-3fae-4178-a1e4-1db655f34a7d",
    title: "Technical Specialist Level 1 - Vietnam",
    team: "Sourcing",
    location: "Nam Định",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/89d4cb82-3fae-4178-a1e4-1db655f34a7d",
  },
  {
    id: "c5d5af1c-8eac-4302-a2fc-14aeec89221d",
    title: "Traceability Specialist",
    team: "Sourcing",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c5d5af1c-8eac-4302-a2fc-14aeec89221d",
  },
  {
    id: "c267e6b9-9704-4010-a09f-a72d5483e8a1",
    title: "Vendor Sourcing Specialist",
    team: "Sourcing",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c267e6b9-9704-4010-a09f-a72d5483e8a1",
  },
  {
    id: "db298a47-9935-4a25-b4bc-1bb8ad8e8027",
    title: "Vice President, Information Security",
    team: "Cyber Security",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/db298a47-9935-4a25-b4bc-1bb8ad8e8027",
  },
  {
    id: "01d84ca9-f0b8-4b92-abc8-5407d92b9977",
    title: "Vice President, Women’s Merchandising",
    team: "Merchandising",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/01d84ca9-f0b8-4b92-abc8-5407d92b9977",
  },
  {
    id: "a9deef58-15c9-4189-b85d-a1e75443d167",
    title: "Visual Merchandising Planner",
    team: "Visual Merchandising",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/a9deef58-15c9-4189-b85d-a1e75443d167",
  },
  {
    id: "8a3a0eba-6dd6-4547-a4b6-62b39922af3c",
    title: "Wholesale Operations Specialist - EMEA",
    team: "EMEA",
    location: "Garching, Munich",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/8a3a0eba-6dd6-4547-a4b6-62b39922af3c",
  },
  {
    id: "dac063ab-2738-4a24-93cd-7f18ad2cb9c7",
    title: "Allocations Coordinator - Bilingual",
    team: "Merchandise Planning & Allocations",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/dac063ab-2738-4a24-93cd-7f18ad2cb9c7",
  },
  {
    id: "3c3f8f0a-1431-4cb6-8d7c-56a806e91edf",
    title: "Business Analyst, Data & Analytics",
    team: "Data & Analytics",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/3c3f8f0a-1431-4cb6-8d7c-56a806e91edf",
  },
  {
    id: "10404f35-0784-4b56-ae28-77dfe948eb4d",
    title: "Community Marketing Manager – Germany",
    team: "EMEA",
    location: "Garching, Munich",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/10404f35-0784-4b56-ae28-77dfe948eb4d",
  },
  {
    id: "7c1bf605-2ced-4a5a-b293-c0555d22f752",
    title: "Construction Project Manager",
    team: "Store Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/7c1bf605-2ced-4a5a-b293-c0555d22f752",
  },
  {
    id: "84169b1e-0ef4-4ded-921e-84e20da8662f",
    title: "Directeur régional – Paris | Regional Manager - Paris",
    team: "EMEA",
    location: "Paris, France",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/84169b1e-0ef4-4ded-921e-84e20da8662f",
  },
  {
    id: "2179de02-6e3a-4710-9378-01afe255075a",
    title: "Director, Design – Tokyo Design Center",
    team: "Tokyo Design Centre",
    location: "Tokyo",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/2179de02-6e3a-4710-9378-01afe255075a",
  },
  {
    id: "ebe82030-fdd5-4457-83e7-8b09e907515a",
    title: "Director, Global Footwear Product",
    team: "Footwear",
    location: "Portland, OR",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/ebe82030-fdd5-4457-83e7-8b09e907515a",
  },
  {
    id: "7d8b93b7-cb60-4f36-8527-b189c8f41791",
    title: "Director, Merchandising - Footwear",
    team: "Merchandising",
    location: "Portland, OR",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/7d8b93b7-cb60-4f36-8527-b189c8f41791",
  },
  {
    id: "0f03dc74-419d-4929-a02e-95cafe82b622",
    title: "Director, People & Culture Business Partnerships - Supply Chain",
    team: "Business Partnership",
    location: "North Vancouver, BC (Corporate)",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/0f03dc74-419d-4929-a02e-95cafe82b622",
  },
  {
    id: "74042140-a865-4e38-9da2-b5af6fa2542c",
    title: "Director, Retail Stores - Canada",
    team: "North America Commercial",
    location: "Vancouver, BC",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/74042140-a865-4e38-9da2-b5af6fa2542c",
  },
  {
    id: "19d259c9-d981-4d58-b0da-ec8743c3a653",
    title: "Director, Retail Stores - US East",
    team: "North America Commercial",
    location: "New York, NY",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/19d259c9-d981-4d58-b0da-ec8743c3a653",
  },
  {
    id: "cbe6c499-ff3c-458f-bcf0-fd2ba9f26c97",
    title: "Director, Retail Stores - US West",
    team: "North America Commercial",
    location: "Seattle, WA",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/cbe6c499-ff3c-458f-bcf0-fd2ba9f26c97",
  },
  {
    id: "b8c13e5f-f0c9-415c-9537-e6e2131f866c",
    title: "Executive Assistant",
    team: "Footwear",
    location: "Portland, OR",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/b8c13e5f-f0c9-415c-9537-e6e2131f866c",
  },
  {
    id: "8e95691f-c1f5-4e0c-8956-41a7e8f18e47",
    title: "Executive Assistant - EMEA",
    team: "EMEA",
    location: "London, England",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/8e95691f-c1f5-4e0c-8956-41a7e8f18e47",
  },
  {
    id: "84c3e844-00db-4897-86a3-e0b2648f0e31",
    title: "Female Fitting Model - Size 4-Small - Casual",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/84c3e844-00db-4897-86a3-e0b2648f0e31",
  },
  {
    id: "64c5c8ca-c905-4741-b069-cd0ce9ad0111",
    title: "Forecast Analyst - EMEA",
    team: "EMEA",
    location: "Garching, Munich",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/64c5c8ca-c905-4741-b069-cd0ce9ad0111",
  },
  {
    id: "e2a0dc4d-70cc-4dd0-9dcb-af5be840d14b",
    title: "Guest Services Representative - Temporary",
    team: "Guest Services",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "FT Temp/Seasonal",
    url: "https://jobs.lever.co/arcteryx.com/e2a0dc4d-70cc-4dd0-9dcb-af5be840d14b",
  },
  {
    id: "1bf58f91-24c5-417b-bae4-7be84be69157",
    title: "Guest Services Representative – Bilingual – Temporary",
    team: "Guest Services",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "FT Temp/Seasonal",
    url: "https://jobs.lever.co/arcteryx.com/1bf58f91-24c5-417b-bae4-7be84be69157",
  },
  {
    id: "0e897e35-4927-44c2-9ebe-1f2beeae72cf",
    title: "Internal Control and Risk Management Manager",
    team: "Finance",
    location: "North Vancouver, BC (Corporate)",
    department: "Finance",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/0e897e35-4927-44c2-9ebe-1f2beeae72cf",
  },
  {
    id: "704af038-1a01-4b3a-9fb0-98bbb72b1605",
    title: "Male Fitting Model - Size Large - Casual",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Contractor",
    url: "https://jobs.lever.co/arcteryx.com/704af038-1a01-4b3a-9fb0-98bbb72b1605",
  },
  {
    id: "de3d3566-a8ec-4370-90f9-c28505ec9e00",
    title: "Male Fitting Model - Size Medium - Casual",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Contractor",
    url: "https://jobs.lever.co/arcteryx.com/de3d3566-a8ec-4370-90f9-c28505ec9e00",
  },
  {
    id: "8a148005-a446-4022-8a40-57b24e1d13f8",
    title: "Manager, Delivery and Transportation",
    team: "Warehousing & Logistics",
    location: "New Westminster, BC",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/8a148005-a446-4022-8a40-57b24e1d13f8",
  },
  {
    id: "803e03d2-85c5-41c3-b21d-ba6a2a3b02d7",
    title: "Manager, Inventory Accounting",
    team: "Finance",
    location: "North Vancouver, BC (Corporate)",
    department: "Finance",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/803e03d2-85c5-41c3-b21d-ba6a2a3b02d7",
  },
  {
    id: "ec3e4566-916c-4bd4-88ec-a71b73171345",
    title: "Manager, Merchandising - EMEA",
    team: "EMEA",
    location: "Garching, Munich",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/ec3e4566-916c-4bd4-88ec-a71b73171345",
  },
  {
    id: "372b9c25-6c53-40e5-88a9-3b3b995b8f65",
    title: "Manager, Merchandising – Hardgoods & Accessories",
    team: "Merchandising",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/372b9c25-6c53-40e5-88a9-3b3b995b8f65",
  },
  {
    id: "148ba778-eead-4cf1-b901-ea7d3de5dc2a",
    title: "Manager, Raw Materials Sourcing",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/148ba778-eead-4cf1-b901-ea7d3de5dc2a",
  },
  {
    id: "3b57a6ba-6be0-43af-9a07-b1c06af928eb",
    title: "Materials Developer 1, Trims",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/3b57a6ba-6be0-43af-9a07-b1c06af928eb",
  },
  {
    id: "b40def81-af93-42e7-b06a-e9493a8cfa24",
    title: "Materials Developer 2 - Temporary",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "FT Temp/Seasonal",
    url: "https://jobs.lever.co/arcteryx.com/b40def81-af93-42e7-b06a-e9493a8cfa24",
  },
  {
    id: "4b037e4f-f82a-452f-ac45-255e4f14ab07",
    title: "Materials Developer 2, Hardgoods",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/4b037e4f-f82a-452f-ac45-255e4f14ab07",
  },
  {
    id: "e6b07337-fc8d-4ae7-a34d-9a5e2d13a4eb",
    title: "Merchandise Planner - A",
    team: "Merchandise Planning & Allocations",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/e6b07337-fc8d-4ae7-a34d-9a5e2d13a4eb",
  },
  {
    id: "3dd83433-4285-43da-bab3-29747b857209",
    title: "Merchandise Planner - B",
    team: "Merchandise Planning & Allocations",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/3dd83433-4285-43da-bab3-29747b857209",
  },
  {
    id: "77b0ea7f-75d8-45c3-aee3-80cca6e5b3ed",
    title: "Omnichannel Project Specialist",
    team: "Retail Strategic Initiatives",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/77b0ea7f-75d8-45c3-aee3-80cca6e5b3ed",
  },
  {
    id: "03f74d47-7405-443f-874c-d82c1ec23251",
    title: "Omnichannel Project Specialist - Temporary",
    team: "Retail Strategic Initiatives",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "FT Temp/Seasonal",
    url: "https://jobs.lever.co/arcteryx.com/03f74d47-7405-443f-874c-d82c1ec23251",
  },
  {
    id: "de520532-7b2c-46b0-ae48-443efb6caecd",
    title: "Pattern Maker/Developer 2",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/de520532-7b2c-46b0-ae48-443efb6caecd",
  },
  {
    id: "c17b1655-be45-45bc-9acf-256f600d64ed",
    title: "Payroll Administrator",
    team: "Finance",
    location: "North Vancouver, BC (Corporate)",
    department: "Finance",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c17b1655-be45-45bc-9acf-256f600d64ed",
  },
  {
    id: "e4228da0-da53-4524-a3c1-f4c56aed8f67",
    title: "People & Culture Business Partner – Corporate Centers",
    team: "Business Partnership",
    location: "North Vancouver, BC (Corporate)",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/e4228da0-da53-4524-a3c1-f4c56aed8f67",
  },
  {
    id: "7b3bf6e6-142b-49fd-8b02-65f7b249ac5b",
    title: "People & Culture Generalist - Manufacturing",
    team: "Business Partnership",
    location: "New Westminster, BC",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/7b3bf6e6-142b-49fd-8b02-65f7b249ac5b",
  },
  {
    id: "c7fc2a2c-aeeb-44b4-906a-efdaa2155a86",
    title: "Product Developer 2 - Hardgoods",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c7fc2a2c-aeeb-44b4-906a-efdaa2155a86",
  },
  {
    id: "25160d91-b984-4f18-8572-35a1a88002c7",
    title: "Product Operations Manager - Footwear",
    team: "Product Operations",
    location: "Portland, OR",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/25160d91-b984-4f18-8572-35a1a88002c7",
  },
  {
    id: "83e443e4-4cad-4196-bbaf-15c72107624e",
    title: "Product Owner, Omnichannel POS",
    team: "Retail Strategic Initiatives",
    location: "North Vancouver, BC (Corporate)",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/83e443e4-4cad-4196-bbaf-15c72107624e",
  },
  {
    id: "657453cb-3f00-4c7e-90d8-b5f4fd49348c",
    title: "Quality Specialist - Danang, Vietnam",
    team: "Sourcing",
    location: "Danang",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/657453cb-3f00-4c7e-90d8-b5f4fd49348c",
  },
  {
    id: "5b742ddf-a0f0-4b93-84db-97cc94c1d623",
    title: "Quality Specialist - Hambantota, Sri Lanka",
    team: "Sourcing",
    location: "Hambantota",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/5b742ddf-a0f0-4b93-84db-97cc94c1d623",
  },
  {
    id: "d97234fc-c6d6-4275-ab93-8f6c55d90dd1",
    title: "Quality Specialist - Jakarta, Indonesia",
    team: "Sourcing",
    location: "Jakarta",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/d97234fc-c6d6-4275-ab93-8f6c55d90dd1",
  },
  {
    id: "e04ba67a-1b93-469a-9f2c-fe3c792e17c7",
    title: "Quality Specialist - Shenzhen, China",
    team: "Sourcing",
    location: "Shenzhen",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/e04ba67a-1b93-469a-9f2c-fe3c792e17c7",
  },
  {
    id: "c4fd5fff-0bca-431f-bf7f-dadefffa1dfc",
    title: "Quality Specialist - Vietnam North",
    team: "Sourcing",
    location: "Vietnam North",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c4fd5fff-0bca-431f-bf7f-dadefffa1dfc",
  },
  {
    id: "9fcacf06-8f56-4341-a1d5-fd863cf56cca",
    title: "Raw Materials Sourcing Specialist",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/9fcacf06-8f56-4341-a1d5-fd863cf56cca",
  },
  {
    id: "0223d917-8273-4459-87a0-29d2c63496ac",
    title: "Regional Manager, Central - EMEA",
    team: "EMEA",
    location: "Garching, Munich",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/0223d917-8273-4459-87a0-29d2c63496ac",
  },
  {
    id: "6844fe12-d5df-4d3f-980f-452fe8488872",
    title: "Regional Manager, NYC",
    team: "North America Commercial",
    location: "New York, NY",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/6844fe12-d5df-4d3f-980f-452fe8488872",
  },
  {
    id: "5ab3affc-884d-4237-aad7-082d0bb436d0",
    title: "Regional Manager, Tri-State",
    team: "North America Commercial",
    location: "USA East Coast",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/5ab3affc-884d-4237-aad7-082d0bb436d0",
  },
  {
    id: "5740c50e-85c5-4bc0-9954-70b1b331119f",
    title: "Senior Business Analyst, ERP (SAP)",
    team: "ERP & Business Applications",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/5740c50e-85c5-4bc0-9954-70b1b331119f",
  },
  {
    id: "1f85ca49-e598-46da-8c6f-e07849a67019",
    title: "Senior Compensation Analyst – Retail",
    team: "Total Rewards & People Operations",
    location: "North Vancouver, BC (Corporate)",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/1f85ca49-e598-46da-8c6f-e07849a67019",
  },
  {
    id: "ee43b7bc-745c-4768-a7d7-d077fc6ad8d6",
    title: "Senior Design Developer – Men’s Apparel",
    team: "Product Design",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/ee43b7bc-745c-4768-a7d7-d077fc6ad8d6",
  },
  {
    id: "6ee78b1e-711a-42b3-8aa7-2adfc0fa0fb6",
    title: "Senior Designer - Footwear",
    team: "Footwear",
    location: "Portland, OR",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/6ee78b1e-711a-42b3-8aa7-2adfc0fa0fb6",
  },
  {
    id: "68879563-ffdd-42d5-95e8-53d00335a192",
    title: "Senior Manager, People & Culture Business Partner - Temporary",
    team: "Business Partnership",
    location: "North Vancouver, BC (Corporate)",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/68879563-ffdd-42d5-95e8-53d00335a192",
  },
  {
    id: "410e0ea0-d706-4e50-b7f6-30e54cb11706",
    title: "Senior Manager, People & Culture Business Partnerships – Arc’One",
    team: "Business Partnership",
    location: "New Westminster, BC",
    department: "People & Culture",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/410e0ea0-d706-4e50-b7f6-30e54cb11706",
  },
  {
    id: "1c84f4d8-a4b4-411f-9ee3-0092f72d4020",
    title: "Senior Material Quality Specialist, Footwear",
    team: "Sourcing",
    location: "Dongguan",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/1c84f4d8-a4b4-411f-9ee3-0092f72d4020",
  },
  {
    id: "4d0eea0e-1c1d-4aa6-a77d-950807f48420",
    title: "Senior Materials Developer",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/4d0eea0e-1c1d-4aa6-a77d-950807f48420",
  },
  {
    id: "dc0b3ddf-da31-469a-88a6-2b06a6b22605",
    title: "Senior Pattern Maker",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/dc0b3ddf-da31-469a-88a6-2b06a6b22605",
  },
  {
    id: "d8a84433-8525-41f9-b6c7-e1b8d5cf0914",
    title: "Senior Project Manager, IT PMO",
    team: "Cyber Security",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/d8a84433-8525-41f9-b6c7-e1b8d5cf0914",
  },
  {
    id: "b2875df8-b553-486a-a233-e03b5e835ba8",
    title: "Senior SAP Project Manager",
    team: "ERP & Business Applications",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/b2875df8-b553-486a-a233-e03b5e835ba8",
  },
  {
    id: "59bbe9e3-b744-490e-8ddb-233a7beef788",
    title: "Senior Sourcing Specialist, Raw Materials",
    team: "Raw Materials",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/59bbe9e3-b744-490e-8ddb-233a7beef788",
  },
  {
    id: "0bfbc194-c1aa-4c31-a855-40d82ec56b88",
    title: "Senior Technical Developer, Tokyo Creation Center",
    team: "Product Creation Development",
    location: "Tokyo",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/0bfbc194-c1aa-4c31-a855-40d82ec56b88",
  },
  {
    id: "44326403-ce2e-4859-9a84-e1a405500b12",
    title: "Software Quality Analyst",
    team: "IS Operations",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/44326403-ce2e-4859-9a84-e1a405500b12",
  },
  {
    id: "8cd16968-8418-4a85-9150-fdaacbd998f6",
    title: "Sourcing Manager - Hong Kong",
    team: "Sourcing",
    location: "Hong Kong",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/8cd16968-8418-4a85-9150-fdaacbd998f6",
  },
  {
    id: "90500af5-13c3-4d3f-959a-550f817cd068",
    title: "Studio Manager, Tokyo Creation Center",
    team: "Tokyo Design Centre",
    location: "Tokyo",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/90500af5-13c3-4d3f-959a-550f817cd068",
  },
  {
    id: "c6ce5080-c864-45f7-ad64-2c75f406b6e0",
    title: "Supply Planner",
    team: "Sourcing",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c6ce5080-c864-45f7-ad64-2c75f406b6e0",
  },
  {
    id: "05687487-b72c-4635-92c9-d50d58b4d365",
    title: "Systems & Process Specialist – Global Planning",
    team: "Product Operations",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/05687487-b72c-4635-92c9-d50d58b4d365",
  },
  {
    id: "03dafc0e-de12-41e5-bb1a-b7d37df21070",
    title: "Technical Architect",
    team: "Architecture",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/03dafc0e-de12-41e5-bb1a-b7d37df21070",
  },
  {
    id: "2afef199-c1a8-45b2-b2ae-beb481152e57",
    title: "Technical Developer 2",
    team: "Product Creation Development",
    location: "North Vancouver, BC (Corporate)",
    department: "Product Creation",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/2afef199-c1a8-45b2-b2ae-beb481152e57",
  },
  {
    id: "03ac6831-844b-4d41-b965-4662aac913dd",
    title: "Technical Specialist Level 1 - China",
    team: "Sourcing",
    location: "China North",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/03ac6831-844b-4d41-b965-4662aac913dd",
  },
  {
    id: "89d4cb82-3fae-4178-a1e4-1db655f34a7d",
    title: "Technical Specialist Level 1 - Vietnam",
    team: "Sourcing",
    location: "Nam Định",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/89d4cb82-3fae-4178-a1e4-1db655f34a7d",
  },
  {
    id: "c5d5af1c-8eac-4302-a2fc-14aeec89221d",
    title: "Traceability Specialist",
    team: "Sourcing",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c5d5af1c-8eac-4302-a2fc-14aeec89221d",
  },
  {
    id: "c267e6b9-9704-4010-a09f-a72d5483e8a1",
    title: "Vendor Sourcing Specialist",
    team: "Sourcing",
    location: "North Vancouver, BC (Corporate)",
    department: "Supply Chain",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/c267e6b9-9704-4010-a09f-a72d5483e8a1",
  },
  {
    id: "db298a47-9935-4a25-b4bc-1bb8ad8e8027",
    title: "Vice President, Information Security",
    team: "Cyber Security",
    location: "North Vancouver, BC (Corporate)",
    department: "Information Systems",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/db298a47-9935-4a25-b4bc-1bb8ad8e8027",
  },
  {
    id: "01d84ca9-f0b8-4b92-abc8-5407d92b9977",
    title: "Vice President, Women’s Merchandising",
    team: "Merchandising",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/01d84ca9-f0b8-4b92-abc8-5407d92b9977",
  },
  {
    id: "a9deef58-15c9-4189-b85d-a1e75443d167",
    title: "Visual Merchandising Planner",
    team: "Visual Merchandising",
    location: "North Vancouver, BC (Corporate)",
    department: "Product",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/a9deef58-15c9-4189-b85d-a1e75443d167",
  },
  {
    id: "8a3a0eba-6dd6-4547-a4b6-62b39922af3c",
    title: "Wholesale Operations Specialist - EMEA",
    team: "EMEA",
    location: "Garching, Munich",
    department: "Commercial",
    commitment: "Full-time",
    url: "https://jobs.lever.co/arcteryx.com/8a3a0eba-6dd6-4547-a4b6-62b39922af3c",
  },
];
