// @arcteryx/design-system/src/components/molecules/textImage/__mocks__/textBlocks.ts
import { type RichTextProps } from "../../richText/RichText.types";

export const textItem1: RichTextProps = {
  content: [
    {
      _type: "block",
      children: [
        {
          _key: "a52eea69bf350",
          _type: "span",
          marks: [],
          text: "IS THIS YOU?",
        },
      ],
      markDefs: [],
      style: "h2",
    },
    {
      _type: "block",
      children: [
        {
          _key: "afcf5165068c",
          _type: "span",
          marks: [],
          text: "You are the face of our brand - connecting with long-time and new guests, providing an incredible guest experience ensuring they get into the right products, and feel supported and empowered. You help fulfil our purpose at Arc’teryx to inspire optimism by connecting people with the experience of nature. You are adventure-positive, curious and fuelled by a love of fresh air. You instinctively make people feel welcome. We think of you as a Guide. We build entire marketing campaigns around how genuine, helpful and knowledgeable you are.",
        },
      ],
      markDefs: [],
      style: "normal",
    },
  ],
  textAlign: "left",
  spacing: "1.5rem",
};

export const textItem2: RichTextProps = {
  content: [
    {
      _type: "block",
      children: [
        {
          _type: "span",
          marks: [],
          text: "OUR HIRING PROCESS",
        },
      ],
      markDefs: [],
      style: "h2",
    },
    {
      _type: "block",
      children: [
        {
          _type: "span",
          marks: [],
          text: "We are meticulous about the creation of our products. That same care and attention also informs our quest to land the right people in the right roles. Our team personally reviews all applications to find highly skilled people to join the Arc’teryx team. We believe in human-centred design. And a human recruiting experience. If your application stands out, we’ll be in touch to dig in on your background and clarify that we are the right fit for each other. We look for people who are excited to represent the brand, who reflect our values, and who care to create exceptional experiences.",
        },
      ],
      markDefs: [],
      style: "normal",
    },
  ],
  textAlign: "left",
  spacing: "1.5rem",
};
