// @arcteryx/components-alerts-overlays/src/components/Alert.js
import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const AlertStyles = styled.div`
  font-family: var(--font-urw-din);
  font-size: 15px;
  line-height: 1.5em;
  min-height: 60px;
  display: flex;
  flex-direction: row;
  position: relative;
  justify-content: flex-start;
  align-items: flex-start;
  padding: 0 1rem;
  > span {
    margin-left: 27.5px;
    padding: 18px 0;
    dispaly: table-cell;
    vertical-align: center;
  }
  p {
    margin: 0 0 1.5rem 0 !important;
    max-width: none !important;

    &:last-of-type {
      margin-bottom: 0 !important;
    }
  }
  &:before {
    display: inline-block;
    width: 20px;
    height: 20px;
    position: absolute;
    top: 30px;
    transform: translateY(-50%);
  }
  ${({ type }) => {
    if (type === "info") {
      return `
        background: var(--background-blue);
        border: 1px solid var(--focus-blue);
        color: var(--info-blue);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Information_Icon_Blue.svg);
        }
        p, a {
          color: var(--info-blue) !important;
        }
      `;
    }
    if (type === "caution") {
      return `
        background: var(--background-yellow);
        border: 1px solid var(--medium-yellow);
        color: var(--caution-yellow);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Caution_Icon.svg);
        }
        p, a {
          color: var(--caution-yellow) !important;
        }
      `;
    }
    if (type === "warning") {
      return `
        background: var(--background-red);
        border: 1px solid var(--medium-red);
        color: var(--warning-red);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Warning_Icon.svg);
        }
        p, a {
          color: var(--warning-red) !important;
        }
      `;
    }
    /* istanbul ignore else */
    if (type === "success") {
      return `
        background: var(--background-green);
        border: 1px solid var(--medium-green);
        color: var(--success-green);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Success_Icon.svg);
        }
        p, a {
          color: var(--success-green) !important;
        }
      `;
    }
    /* istanbul ignore else */
    if (type === "advisory") {
      return `
        background: var(--colour-grey-background);
        border: 1px solid  var(--warning-red);
        color: var(--warning-red);
        &:before {
          content: url(//images.arcteryx.com/foundation-ui/svgs/Information_Icon_Red.svg);
        }
        p, a {
          color: var(--warning-red) !important;
        }
      `;
    }
  }}
`;
const Alert = ({ type, dangerouslySetInnerHTML, children, className }) => {
  if (type) {
    return (
      <AlertStyles className={className} type={type}>
        {dangerouslySetInnerHTML ? (
          <span dangerouslySetInnerHTML={{ __html: children }}></span>
        ) : (
          <span>{children}</span>
        )}
      </AlertStyles>
    );
  }
  return null;
};

Alert.defaultProps = {
  type: "info",
};

Alert.prototype = {
  type: PropTypes.string.isRequired,
};

export default Alert;
