// @arcteryx/design-system/src/components/organisms/carousel/components/CarouselSliderControls.tsx
import { Icon } from "@arcteryx/components-icons";
import { P, ScreenSize } from "@arcteryx/components-typography";
import React from "react";
import styled from "styled-components";

const StyledButtonDiv = styled.div`
  display: flex;
  align-self: center;
  width: 350px;
  align-items: center;
  justify-content: space-between;
  @media (min-width: ${ScreenSize.lg}px) {
    display: none;
  }
`;

const StyledButton = styled.button`
  background: none;
  border: none;
  padding: 0;
  margin: 0;
  cursor: pointer;
`;

interface Props {
  isPrevButtonDisabled: boolean;
  isNextButtonDisabled: boolean;
  currentIndex: number;
  itemCount: number;
  onClickPrevious: () => void;
  onClickNext: () => void;
}

const Button: React.FC<{
  isEnabled: boolean;
  onClick: () => void;
  icon: "left-arrow" | "right-arrow";
}> = ({ isEnabled, onClick, icon }) => {
  return isEnabled ? (
    <StyledButton onClick={onClick}>
      <Icon type={icon} height={"30"} width={"30"} />
    </StyledButton>
  ) : (
    <StyledButton disabled={!isEnabled}>
      <Icon type={icon} height={"30"} width={"30"} color="grey" />
    </StyledButton>
  );
};

export const CarouselSliderControls: React.FC<Props> = ({
  isNextButtonDisabled,
  isPrevButtonDisabled,
  currentIndex,
  itemCount,
  onClickNext,
  onClickPrevious,
}) => {
  return (
    <StyledButtonDiv>
      <Button
        isEnabled={!isPrevButtonDisabled}
        onClick={onClickPrevious}
        icon="left-arrow"
      />
      <P>
        {" "}
        {currentIndex} / {itemCount}
      </P>
      <Button
        isEnabled={!isNextButtonDisabled}
        onClick={onClickNext}
        icon="right-arrow"
      />
    </StyledButtonDiv>
  );
};
