// @arcteryx/design-system/src/components/organisms/mediumHeroBanner/MediumHeroBanner.tsx
import React from "react";
import styled, { css } from "styled-components";
import { RichText } from "../../molecules";
import { type MediumHeroBannerProps } from "./MediumHeroBanner.types";
import { ScreenSize } from "@arcteryx/components-typography";

const mediumHeroBannerDimensions = {
  xs: { width: 375, height: 112 },
  sm: { width: 576, height: 172 },
  md: { width: 768, height: 230 },
  lg: { width: 992, height: 360 },
  xl: { width: 1200, height: 360 },
  xxl: { width: 1440, height: 460 },
};

const MediumHeroWidthAndHeightStyles = css`
  height: 90vh;
  max-height: ${mediumHeroBannerDimensions.xs.height}px;
  @media (min-width: ${ScreenSize.sm}px) {
    max-height: ${mediumHeroBannerDimensions.sm.height}px;
  }

  @media (min-width: ${ScreenSize.md}px) {
    max-height: ${mediumHeroBannerDimensions.md.height}px;
  }

  @media (min-width: ${ScreenSize.xl}px) {
    max-height: ${mediumHeroBannerDimensions.xl.height}px;
  }

  @media (min-width: ${ScreenSize.xxl}px) {
    max-height: ${mediumHeroBannerDimensions.xxl.height}px;
  }
`;

const StyledMediumHeroBannerContainer = styled.div<{ $image: string }>`
  ${MediumHeroWidthAndHeightStyles};
  width: 100%;
  overflow: hidden;
  display: block !important;
  background-image: url(${(props) => props.$image});
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
`;

const StyledTextOverlay = styled.div`
  ${MediumHeroWidthAndHeightStyles};
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const StyledRichText = styled.div`
  letter-spacing: 0.45px;
`;

const zero = "0rem";

export const MediumHeroBanner: React.FC<MediumHeroBannerProps> = ({
  image,
  richText,
}) => {
  return (
    <StyledMediumHeroBannerContainer
      $image={image}
      data-testid="med-hero-background"
    >
      <StyledTextOverlay>
        <StyledRichText>
          <RichText content={richText} spacing={zero} />
        </StyledRichText>
      </StyledTextOverlay>
    </StyledMediumHeroBannerContainer>
  );
};
