// @arcteryx/components-button/src/Button.stories.js
import React, { useState } from "react";
import Button from "./components/button";
import { action } from "@storybook/addon-actions";
import styled from "styled-components";
import { Icon } from "@arcteryx/components-icons";

const contextOptions = {
  Standard: "Standard",
  Inverted: "Inverted",
  Subtle: "Subtle",
  Secondary: "Secondary-cta",
  Secure: "Secure-Checkout",
  AddToCart: "Add-To-Cart",
  Email: "Email",
};
const Front = styled.div`
  width: ${props => props.width || "100%"};
  height: 100%;
  display: block;
  padding: 10px 0;
  border: 1px solid #ccc;
  margin: 0 auto;
  text-align: center;
`;
const ButtonWithWaiting = props => {
  const [isWaiting, setIsWaiting] = useState(false);

  const handleClick = () => {
    setIsWaiting(true);
  };

  return (
    <Button context={props.context} isWaiting={isWaiting} onClick={handleClick}>
      {props.children}
    </Button>
  );
};

export default {
  title: "Button",
  component: Button,
};

export const ControlledButton = ({ children: label, ...args }) => (
  <Button onClick={action("button_clicked")} {...args}>
    {label}
  </Button>
);

ControlledButton.args = {
  children: "Button",
};

ControlledButton.argTypes = {
  context: {
    options: contextOptions,
    control: {
      type: "radio",
    },
  },
  children: {
    name: "Label",
    control: {
      type: "text",
    },
  },
};

export const StandardButton = () => <Button context="Standard">Standard</Button>;
export const StandardButtonIconLeft = () => (
  <Button
    context="Standard"
    IconButtonLeft={<Icon type="product-care" width="20" height="20" color="white" title="cart" iconPosition="left" />}
  >
    Standard Icon left
  </Button>
);
export const StandardButtonIconRight = () => (
  <Button
    context="Standard"
    IconButtonRight={<Icon type="cart" width="20" height="20" color="white" title="cart" iconPosition="right" />}
  >
    Standard Icon Right
  </Button>
);
export const InvertedButton = () => <Button context="Inverted">Inverted Button</Button>;

export const AnchorLinkButton = () => (
  <Button
    onClick={e => e.preventDefault()}
    href="https://www.arcteryx.com"
    target="_blank"
    rel="noopener noreferrer"
    context="Secondary-cta"
    disabled={false}
  >
    Anchor
  </Button>
);
export const ButtonWithLongLabel = () => (
  <Front width="220px">
    <Button context="Standard">Standard Button on Two Lines</Button>
  </Front>
);
export const SubtleButton = () => <Button context="Subtle">Subtle</Button>;
export const SecondaryButton = () => <Button context="Secondary-cta">Secondary</Button>;
export const SecureCheckoutButton = () => <Button context="Secure-Checkout"> Checkout</Button>;
export const AddToCartButton = () => <Button context="Add-To-Cart">Add To Cart</Button>;
export const EmailSignUpButton = () => <Button context="Email">Notify Me</Button>;
export const FindInStoreContext = () => <Button context="Find-In-Store">Find in store</Button>;
export const FindInStore = () => (
  <Button context="Subtle" className="find-in-store">
    Find in store
  </Button>
);
export const StandardDisabledButton = () => (
  <Button context="Standard" disabled>
    Standard
  </Button>
);
export const ExampleButtonsWithWaitingState = () => (
  <>
    <Front>
      <ButtonWithWaiting context="Standard" isWaiting>
        Click Me
      </ButtonWithWaiting>
    </Front>
    <Front>
      <ButtonWithWaiting context="Add-To-Cart" isWaiting>
        Add To Cart
      </ButtonWithWaiting>
    </Front>
    <Front>
      <ButtonWithWaiting context="Subtle" isWaiting>
        Subtle
      </ButtonWithWaiting>
    </Front>
    <Front>
      <ButtonWithWaiting context="Secure-Checkout" isWaiting>
        Checkout
      </ButtonWithWaiting>
    </Front>
  </>
);
