// @arcteryx/design-system/.storybook/preview.tsx
import React from "react";

import type { Preview } from "@storybook/react";
import { GlobalStyles, ScreenSize } from "@arcteryx/components-typography";

const preview: Preview = {
  parameters: {
    viewport: {
      viewports: {
        xs: {
          name: "Mobile (XSmall)",
          styles: {
            width: "375px",
            height: "1024px",
          },
        },
        sm: {
          name: "Mobile (Small)",
          styles: {
            width: `${ScreenSize.sm}px`,
            height: "1024px",
          },
        },
        md: {
          name: "Tablet (Medium)",
          styles: {
            width: `${ScreenSize.md}px`,
            height: "1024px",
          },
        },
        lg: {
          name: "Desktop (Large)",
          styles: {
            width: `${ScreenSize.lg}px`,
            height: "1268px",
          },
        },
        xl: {
          name: "Desktop (X-Large)",
          styles: {
            width: `${ScreenSize.xl}px`,
            height: "1600px",
          },
        },
        "2xl": {
          name: "Desktop (2X-Large)",
          styles: {
            width: "1440px",
            height: "1920px",
          },
        },
      },
    },
    actions: { argTypesRegex: "^on[A-Z].*" },
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/,
      },
    },
  },
  decorators: [
    (Story) => (
      <div>
        <GlobalStyles />
        <Story />
      </div>
    ),
  ],
};

export default preview;
