// @arcteryx/design-system/src/components/organisms/collapseGroup/CollapseGroup.stories.tsx
import type React from "react";
import { CollapseGroup as ArcteryxCollapseGroup } from "./CollapseGroup";
import { type Meta, type StoryObj } from "@storybook/react";
import { mensSizingBlocks } from "../../molecules/card/__mocks__/cardRichText";

type Props = React.ComponentProps<typeof ArcteryxCollapseGroup>;

const meta: Meta<typeof ArcteryxCollapseGroup> = {
  component: ArcteryxCollapseGroup,
  title: "Design System / Organisms / Collapse Group",
};
export default meta;

type Story = StoryObj<typeof ArcteryxCollapseGroup>;

const Args: Props = {
  title: "Collapse Group",
  items: [
    {
      title: "Accordion 1",
      content: mensSizingBlocks,
      headingLevel: "5",
      isOpenOnPageLoad: false,
      large: false,
    },
    {
      title: "Accordion 2",
      content: mensSizingBlocks,
      headingLevel: "3",
      isOpenOnPageLoad: true,
      large: false,
    },
    {
      title: "Accordion 3",
      content: mensSizingBlocks,

      headingLevel: "3",
      isOpenOnPageLoad: false,
      large: false,
      minWidthStayOpen: 900,
    },
  ],
};

export const CollapseGroup: Story = {
  args: Args,
};
