// @arcteryx/design-system/src/components/molecules/tableSizing/__mocks__/tableSizingMocks.ts
import { type TableSizingProps } from "..";

export const inchesTable: TableSizingProps = {
  description: "PACK SIZING, Your body measurements in inches:",
  measuringType: "inches",
  table: [
    ["Size", "Sleeve", "Chest", "Waist", "Hip"],
    ["XXS", "30", "33", "27", "32 ½"],
    ["XS", "31 ½", "35", "29", "34 ½"],
    ["S", "32 ½", "37", "31", "36 ½"],
    ["M", "34", "40", "34", "39 ½"],
    ["L", "35", "43", "37", "42 ½"],
    ["XL", "36", "47", "40 ½", "46 ½"],
    ["XXL", "37", "51", "44 ½", "50"],
    ["3XL", "38", "54 ½", "48 ½", "54"],
  ],
};

export const cmTable: TableSizingProps = {
  description: "PACK SIZING, Your body measurements in centimeters:",
  measuringType: "cm",
  table: [
    ["Size", "Sleeve", "Chest", "Waist", "Hip"],
    ["XXS", "78", "84", "69", "83"],
    ["XS", "81", "89", "74", "88"],
    ["S", "83", "94", "79", "93"],
    ["M", "86", "102", "86", "100"],
    ["L", "89", "109", "94", "108"],
    ["XL", "91", "119", "104", "118"],
    ["XXL", "94", "129", "114", "128"],
    ["3XL", "96", "139", "124", "138"],
  ],
};

export const internationalTable: TableSizingProps = {
  description: "PACK SIZING, International size equivalents:",
  measuringType: "international",
  table: [
    ["Size", "Europe", "UK", "China", "Japan", "Korea"],
    ["XXS", "42", "32", "170/86A", "XS", "85"],
    ["XS", "44", "34", "170/92A", "S", "90"],
    ["S", "46", "36", "170/96A", "M", "95"],
    ["M", "48", "38", "175/102A", "L", "100"],
    ["L", "52", "42", "175/110A", "LL", "110"],
    ["XL", "56", "46", "180/118A", "3L", "120"],
    ["XXL", "60", "50", "180/128A", "4L", "130"],
  ],
};

export const footwearSizingTable: TableSizingProps = {
  description: "Your shoe size equivalents:",
  measuringType: "",
  table: [
    ["cm", "US M's", "US W's", "UK", "EU", "Mondo"],
    ["22", "4", "5", "3.5", "36", "22"],
    ["22.5", "4.5", "5.5", "4", "36 ⅔", "22.5"],
    ["23", "5", "6", "4.5", "37 ⅓", "23"],
    ["23.5", "5.5", "6.5", "5", "38", "23.5"],
    ["24", "6", "7", "5.5", "38 ⅔", "24"],
    ["24.5", "6.5", "7.5", "6", "39 ⅓", "24.5"],
    ["25", "7", "8", "6.5", "40", "25"],
    ["25.5", "7.5", "8.5", "7", "40 ⅔", "25.5"],
    ["26", "8", "9", "7.5", "41 ⅓", "26"],
    ["26.5", "8.5", "9.5", "8", "42", "26.5"],
    ["27", "9", "10", "8.5", "42 ⅔", "27"],
    ["27.5", "9.5", "10.5", "9", "43 ⅓", "27.5"],
    ["28", "10", "11", "9.5", "44", "28"],
    ["28.5", "10.5", "11.5", "10", "44 ⅔", "28.5"],
    ["29", "11", "12", "10.5", "45 ⅓", "29"],
    ["29.5", "11.5", "12.5", "11", "46", "29.5"],
    ["30", "12", "13", "11.5", "46 ⅔", "30"],
    ["30.5", "12.5", "13.5", "12", "47 ⅓", "30.5"],
    ["31", "13", "14", "12.5", "48", "31"],
    ["31.5", "13.5", "14.5", "13", "48 ⅔", "31.5"],
    ["32", "14", "15", "13.5", "49 ⅓", "32"],
  ],
};
