// @arcteryx/design-system/src/components/atoms/tabs/Tabs.stories.tsx
import React, { useState } from "react";
import { type Meta, type StoryObj } from "@storybook/react";
import { TabButton, TabList } from "@arcteryx/components-tabs";
import styled from "styled-components";

const StyledContainer = styled.div`
  width: 70%;
  margin: 0 auto;
  padding: 1rem 0;
`;

const initialProps = {
  tabs: [
    {
      label: "Sizing Chart",
    },
    {
      label: "Fit Guide",
    },
    {
      label: "Layering",
    },
  ],
};

const meta: Meta<typeof TabList> = {
  component: TabList,
  title: "Design System / Atoms / Tab",
  args: initialProps,
};

export default meta;

type Story = StoryObj<typeof TabList>;
type tabArgs = Story["args"];

export const Sizing: Story = {
  render: (args: tabArgs) => {
    const [tabIndex, setTabIndex] = useState(0);
    return (
      <StyledContainer>
        <TabList selectedIndex={tabIndex} setSelectedIndex={setTabIndex}>
          {initialProps.tabs?.map((tab: any) => (
            <TabButton key={tab.label} href={tab.url}>
              {tab.label}
            </TabButton>
          ))}
        </TabList>
      </StyledContainer>
    );
  },
  args: initialProps,
};
