// @arcteryx/components-accordion/src/components/RadioAccordion.styles.tsx
import styled, { css } from 'styled-components';

export const AccordionContainer = styled.div`
  border-top: 1px solid var(--colour-grey-pinline);
`;

export const AccordionItemStyled = styled.div`
  border-bottom: 1px solid var(--colour-grey-pinline);
  &:last-child {
    border-bottom: 1px solid var(--colour-grey-pinline);
  }
`;

export const CustomRadioButton = styled.div<{ isSelected: boolean }>`
  position: relative;
  margin-right: 0.6rem;
  border: 1px solid var(--colour-grey-pinline);
  border-radius: 50%;
  height: 1rem;
  width: 1rem;
  transition: border-color 0.2s ease-out, box-shadow 0.2s ease-out;

  &:after {
    content: '';
    position: absolute;
    margin: 0 auto;
    left: 50%;
    top: 50%;
    height: 0.7rem;
    width: 0.7rem;
    background-color: var(--colour-black);
    border-radius: 50%;
    transform: translate(-50%, -50%) scale(0);
    transition: transform 0.3s ease-out;
  }

  ${({ isSelected }) =>
    isSelected &&
    css`
      &:after {
        transform: translate(-50%, -50%) scale(1);
      }
    `}
`;

export const RadioInput = styled.input`
  opacity: 0;
  position: absolute;
  margin: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
`;

export const InputLabelContainer = styled.label`
  display: flex;
  align-items: center;
  cursor: pointer;
  width: 100%;
  margin: 0.6rem 0;
`;

export const AccordionContent = styled.div<{isVisible: boolean}>`
  display: none;
  padding: 0.6rem;
  margin-left: 1rem;
  ${({ isVisible }) =>
    isVisible &&
    css`
      display: block;
    `}
`;
