// @arcteryx/design-system/src/components/organisms/mediumHeroBanner/__mocks__/richTextBlocks.ts
import { type RichTextBlocks } from "../../../molecules";

export const RichTextBlocksWithAlternatingColors: RichTextBlocks = [
  {
    _key: "987f0cc2eeb5",
    _type: "block",
    children: [
      {
        _key: "0504462513fa",
        _type: "span",
        marks: ["96e0712f0b15"],
        text: "Best.",
      },
    ],
    markDefs: [
      {
        _key: "96e0712f0b15",
        _type: "textColor",
        label: "White",
        value: "#FFFFFF",
      },
    ],
    style: "h1",
  },
  {
    _key: "8acec025b450",
    _type: "block",
    children: [
      {
        _key: "43bd88fe6f18",
        _type: "span",
        marks: ["2210cf5a6551"],
        text: "Seams.",
      },
    ],
    markDefs: [
      {
        _key: "2210cf5a6551",
        _type: "textColor",
        label: "Black",
        value: "#000000",
      },
    ],
    style: "h1Alt",
  },
  {
    _key: "5dc9cff6c9af",
    _type: "block",
    children: [
      {
        _key: "89a38c6ed165",
        _type: "span",
        marks: ["ee62f1535397"],
        text: "Ever.",
      },
    ],
    markDefs: [
      {
        _key: "ee62f1535397",
        _type: "textColor",
        label: "White",
        value: "#FFFFFF",
      },
    ],
    style: "h1",
  },
];

export const RichTextBlocksWithDefaultColor: RichTextBlocks = [
  {
    _key: "5dc9cff6c9af",
    _type: "block",
    children: [
      {
        _key: "89a38c6ed165",
        _type: "span",
        marks: [],
        text: "Best.",
      },
    ],
    markDefs: [],
    style: "h1",
  },
  {
    _key: "fbe741960974",
    _type: "block",
    children: [
      {
        _key: "aa3935ace4b2",
        _type: "span",
        marks: [],
        text: "Seams.",
      },
    ],
    markDefs: [],
    style: "h1Alt",
  },
  {
    _key: "a8b97f2e3d7b",
    _type: "block",
    children: [
      {
        _key: "7653b19509bd",
        _type: "span",
        marks: [],
        text: "Ever.",
      },
    ],
    markDefs: [],
    style: "h1",
  },
];
