// @arcteryx/design-system/src/components/atoms/button/Button.stories.tsx
import React from "react";
import { type Meta, type StoryObj } from "@storybook/react";
import { Button as ArcteryxButton } from "@arcteryx/components-button";
import { type ButtonProps, buttonOptions } from "./Button.types";

const defaultButtonProps: ButtonProps = {
  context: "Standard",
  disabled: false,
  onClick: () => {},
  label: "SHOP NOW",
};

const meta: Meta<typeof ArcteryxButton> = {
  component: ArcteryxButton,
  title: "Design System/Atoms/Button",
  args: defaultButtonProps,
  argTypes: {
    isWaiting: {
      control: "boolean",
    },
    onClick: { action: "clicked" },
    onContextMenu: { action: "right-clicked" },
  },
};

export default meta;

type Story = StoryObj<ButtonProps>;

export const Button: Story = {
  render: (args: ButtonProps) => (
    <ArcteryxButton {...args}>{args.label}</ArcteryxButton>
  ),
  args: defaultButtonProps,
  argTypes: {
    context: {
      control: "radio",
      options: buttonOptions,
    },
  },
};

export const Link: Story = {
  args: {
    href: "https://www.arcteryx.com",
    target: "_blank",
    rel: "noopener noreferrer",
    context: "Secondary-cta",
  },
  render: (args: ButtonProps) => (
    <ArcteryxButton {...args}>Link</ArcteryxButton>
  ),
};
