// @arcteryx/design-system/src/components/organisms/video/Video.stories.tsx
import { type Meta, type StoryObj } from "@storybook/react";
import { Video as ArcteryVideo } from "./Video";
import { type VideoProps } from "./Video.types";

const meta: Meta<typeof ArcteryVideo> = {
  component: ArcteryVideo,
  title: "Design System/Organisms/Video",
};
export default meta;

type Story = StoryObj<typeof ArcteryVideo>;

const video: VideoProps = {
  videoSrc: "https://www.youtube.com/embed/Kd3IOqFWhTU?si=JMIfhXKlSz-0TqXx",
  isAutoPlayed: false,
  isMuted: false,
  videoThumbnail: {
    title: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
    subtitle: "13:34",
    isPlayButtonEnabled: true,
    playButtonColour: "#FFFFFF",
    textColour: "#FFFFFF",
    isThumbnailGradientEnabled: false,
    image: {
      src: "https://images-dynamic-arcteryx.imgix.net/campaign/S23/ski-snow/assets/images/panel10_desktop_arcteryx-ski-snow-hub-goggles.jpg?fit=crop&h=1191&w=1055&q=45&fm=webp",
      alt: "How to wash your GORE-TEX",
    },
  },
};

export const Video: Story = {
  args: video,
};
