// @arcteryx/components-alerts-overlays/src/Alerts.stories.js
import React from "react";
import { Alert } from "./index";
import styled from "styled-components";

const Front = styled.div`
  width: 100%;
  height: 100%;
  min-height: 300px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export default {
  title: "Alerts",
  component: Alerts,
};

export const Alerts = () => {
  return (
    <>
      {" "}
      <h1>Alerts</h1>
      <Front>
        <Info />
        <Advisory />
        <Warning />
        <Caution />
        <Success />
      </Front>
    </>
  );
};
export const Info = () => {
  return (
    <>
      <Alert type="info">
        <p>
          Our warehouse will be closed on January 15th, 2020 due to the local mountains receiving 100cm of snow
          overnight. 2020 due to the local mountains receiving 100cm of snow overnight. Our warehouse will be closed on
          January 15th, 2020 due to the local mountains receiving 100cm of snow overnight. 2020 due to the local
          mountains receiving 100cm of snow overnight. Our warehouse will be closed on January 15th, 2020 due to the
          local mountains receiving 100cm of snow overnight. 2020 due to the local mountains receiving 100cm of snow
          overnight.
        </p>

        <p>
          Our warehouse will be closed on January 15th, 2020 due to the local mountains receiving 100cm of snow
          overnight. 2020 due to the local mountains receiving 100cm of snow overnight. Our warehouse will be closed on
          January 15th, 2020 due to the local mountains receiving 100cm of snow overnight. 2020 due to the local
          mountains receiving 100cm of snow overnight. <a href="/">Learn More.</a>
        </p>
      </Alert>
    </>
  );
};

export const Advisory = () => {
  return (
    <>
      <Alert type="advisory">
        COVID-19 Advisory: For the health and safety of our team, we are continuing to service your gear through digital
        assessments at our Product Service Centre. At the moment, only repairs for issues covered by our Limited
        Warranty will be available. Repairs for issues such as wear and tear or cosmetic damage will be unavailable at
        this time. To help us find a solution, please include pictures when submitting your product service request. If
        you have any questions, please give us a call or send a message to one of our representatives.
      </Alert>
    </>
  );
};

export const Caution = () => {
  return (
    <>
      <Alert type="caution" dangerouslySetInnerHTML="true">
        {`Our warehouse will be <em>closed</em> on January 15th, 2020 due to the local mountains receiving 100cm of snow overnight.`}
      </Alert>
    </>
  );
};

export const Warning = () => {
  return (
    <>
      <Alert type="warning">
        Our warehouse will be closed on January 15th, 2020 due to the local mountains receiving 100cm of snow overnight.
      </Alert>
    </>
  );
};

export const Success = () => {
  return (
    <>
      <Alert type="success">
        Our warehouse is open as usual. Return to your regular scheduled shopping experience.
      </Alert>
    </>
  );
};
