// @arcteryx/components-alerts-overlays/src/Toast.stories.js
import React, { useState } from "react";
import { Toast } from "./index";
import styled from "styled-components";
import { TOAST_TYPE } from "./components/Toast.js";

const Front = styled.div`
  width: 100%;
  height: 100%;
  min-height: 300px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export default {
  title: "Toasts",
  component: Toasts,
};

export const Toasts = () => {
  const [flag, setFlag] = useState("");
  return (
    <>
      {" "}
      <h1>Toasts</h1>
      <button
        onClick={() => {
          setFlag(TOAST_TYPE.SUCCESS);
        }}
      >
        Success
      </button>
      <br />
      <button
        onClick={() => {
          setFlag(TOAST_TYPE.WARNING);
        }}
      >
        Warning
      </button>
      <br />
      <button
        onClick={() => {
          setFlag(TOAST_TYPE.CAUTION);
        }}
      >
        Caution
      </button>
      <br />
      <Front>
        {flag === TOAST_TYPE.SUCCESS && <Success />}
        {flag === TOAST_TYPE.WARNING && <Warning />}
        {flag === TOAST_TYPE.CAUTION && <Caution />}
      </Front>
    </>
  );
};

export const Caution = () => {
  return (
    <>
      <Toast type={TOAST_TYPE.CAUTION}>
        Maximum quantity exceeded. <a href="/">Arc'teryx Bulk Buying Policy.</a>
      </Toast>
    </>
  );
};

export const Warning = () => {
  return (
    <>
      <Toast type={TOAST_TYPE.WARNING}>Sorry, there is insufficient inventory to fulfill your request.</Toast>
    </>
  );
};

export const Success = () => {
  return (
    <>
      <Toast type={TOAST_TYPE.SUCCESS}>Email subscription preferences saved.</Toast>
    </>
  );
};
