// @arcteryx/components-form/src/components/Tooltip/styles.js
import styled from "styled-components";

export const Div = styled.div`
  display: inline-block;
  position: relative;
  bottom: -1px;
  padding: 5px;
  height: max(${props => props.size}, 20px);
  color: transparent;

  cursor: pointer;
  &:after {
    content: ${props =>
      props.isInfo === true
        ? "url(//images.arcteryx.com/foundation-ui/svgs/Information_Icon.svg)"
        : "url(//images.arcteryx.com/foundation-ui/svgs/Question_Icon.svg)"};
    width: ${props => props.size};
    display: inline-block;
  }
`;

export const TooltipText = styled.span`
  ${props => !props.displayTooltip && `display: none;`}

  --vWidth: 360px;
  --vMinWidth: min(${props => props.maxWidth}px, var(--vWidth));

  position: absolute;
  width: var(--vWidth);
  background: var(--colour-black);
  color: var(--colour-white);
  box-shadow: 6px 6px 6px #00000029;
  padding: 1rem;
  font-size: 1rem;
  line-height: 1.5rem;
  z-index: 999;

  max-height: ${props => `${props.maxHeight}px`};
  max-width: ${props => `${props.maxWidth}px`};

  ${props =>
    props.arrowDirection === "top" &&
    `
    top: 160%;
    left: 50%;
    margin-left: calc(-1 * var(--vMinWidth) / 2 + ${props.size} / 2);
    &:after {
      top: -6px;
      transform: rotate(180deg);
    }
  `}
  // Defaults to bottom
  ${props =>
    props.arrowDirection === "bottom" &&
    `
    bottom: 110%;
    left: 50%;
    margin-left: calc(-1 * var(--vMinWidth) / 2 + ${props.size} / 2);
    &:after {
      bottom: -6px;
    }
  `}
  ${props =>
    props.arrowDirection === "left" &&
    `
    width: max-content;
    left: 100%;
    &:after {
      top: 4px;
      left: -0.667rem !important;
      transform: rotate(90deg);   
    }
  `}
  ${props =>
    props.arrowDirection === "right" &&
    `
    width: max-content;
    right: 100%;
    &:after {
      top: 4px;
      right: -0.7rem !important;
      transform: rotate(-90deg);
    }
  `}

  &:after {
    position: absolute;
    content: "";
    display: var(--colour-black);
    width: 0;
    height: 0;
    border: 8px solid transparent;
    border-bottom-width: 0;
    border-top-color: var(--colour-black);
    right: 50%;
  }
`;

export const TooltipWrapper = styled.span`
  align-items: center;
  max-width: 360px;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: 1fr auto;
  display: grid;
`;

export const TooltipTipImg = styled.img`
  height: 50px;
  width: 150px;
  position: relative;
  display: inline-block;
  margin: 0.5rem 0;
  background: black;
`;

export const SubHeader = styled.span`
  font-weight: 600;
`;

export const Content = styled.span`
  grid-column: 1/3;
  overflow: hidden;
`;
