// @arcteryx/design-system/src/components/organisms/carousel/Carousel.stories.tsx
import { type Meta, type StoryObj } from "@storybook/react";
import { Carousel as CarouselComponent } from "./Carousel";
import {
  carouselCardStoryMock,
  carouselImageStoryMock,
} from "./__mocks__/carouselMocks";

const meta: Meta<typeof CarouselComponent> = {
  component: CarouselComponent,
  title: "Design System / Organisms / Carousel",
};
export default meta;

type Story = StoryObj<typeof CarouselComponent>;

export const CarouselCardSeven: Story = {
  args: carouselCardStoryMock,
  name: "Carousel Card:  1-7 items",
};

export const CarouselCardEight: Story = {
  args: {
    ...carouselCardStoryMock,
    children: [
      ...carouselCardStoryMock.children,
      carouselCardStoryMock.children[0],
    ],
  },
  name: "Carousel Card: 8+ items",
};

export const CarouselImageSeven: Story = {
  args: carouselImageStoryMock,
  name: "Carousel Image: 1-7 items",
};

export const CarouselImageEight: Story = {
  args: {
    ...carouselImageStoryMock,
    children: [
      ...carouselImageStoryMock.children,
      carouselImageStoryMock.children[0],
    ],
  },
  name: "Carousel Image: 8+ items",
};
