// @arcteryx/components-universal/dist/umd/index.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('react'), require('@arcteryx/components-contexts'), require('styled-components'), require('@arcteryx/components-typography'), require('prop-types'), require('@arcteryx/components-hooks')) :
  typeof define === 'function' && define.amd ? define(['exports', 'react', '@arcteryx/components-contexts', 'styled-components', '@arcteryx/components-typography', 'prop-types', '@arcteryx/components-hooks'], factory) :
  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, (function () {
    var current = global.ARCTERYX && global.ARCTERYX.componentsuniversal;
    var exports = (global.ARCTERYX = global.ARCTERYX || {}, global.ARCTERYX.componentsuniversal = {});
    factory(exports, global.React, global.componentsContexts, global.styled, global.componentsTypography, global.PropTypes, global.componentsHooks);
    exports.noConflict = function () { global.ARCTERYX.componentsuniversal = current; return exports; };
  })());
})(this, (function (exports, React, componentsContexts, styled, componentsTypography, PropTypes, componentsHooks) { 'use strict';

  function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

  var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
  var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);
  var PropTypes__default = /*#__PURE__*/_interopDefaultLegacy(PropTypes);

  function _extends() {
    _extends = Object.assign ? Object.assign.bind() : function (target) {
      for (var i = 1; i < arguments.length; i++) {
        var source = arguments[i];
        for (var key in source) {
          if (Object.prototype.hasOwnProperty.call(source, key)) {
            target[key] = source[key];
          }
        }
      }
      return target;
    };
    return _extends.apply(this, arguments);
  }
  function _objectWithoutPropertiesLoose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for (i = 0; i < sourceKeys.length; i++) {
      key = sourceKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      target[key] = source[key];
    }
    return target;
  }
  function _objectWithoutProperties(source, excluded) {
    if (source == null) return {};
    var target = _objectWithoutPropertiesLoose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
      var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
      for (i = 0; i < sourceSymbolKeys.length; i++) {
        key = sourceSymbolKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
        target[key] = source[key];
      }
    }
    return target;
  }

  const Script = _ref => {
    let {
      url,
      async = true,
      defer = false,
      injectInHead = true,
      WrapComponent = null
    } = _ref;
    React.useEffect(() => {
      if (injectInHead && !WrapComponent) {
        const script = document.createElement("script");
        script.src = url;
        script.async = async;
        script.defer = defer;
        document.head.appendChild(script);
      }
    }, [url, async, injectInHead, WrapComponent, defer]);
    let component = url && (!injectInHead || WrapComponent) ? /*#__PURE__*/React__default["default"].createElement("script", {
      src: url,
      async: async,
      defer: defer
    }) : null;
    if (component && WrapComponent) {
      component = /*#__PURE__*/React__default["default"].createElement(WrapComponent, null, component);
    }
    return component;
  };

  var config = {
    production: {
      launch: {
        outdoor: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-89990b0a95f5.min.js",
        leaf: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-89990b0a95f5.min.js",
        urban: "//assets.adobedtm.com/323c24ea560f/a9ff8797aaf9/launch-1e4fb715a4f8.min.js",
        sale: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-89990b0a95f5.min.js"
      },
      evergage: "//cdn.evgnet.com/beacon/arcteryx/production/scripts/evergage.min.js"
    },
    preprod: {
      launch: {
        outdoor: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-89990b0a95f5.min.js",
        leaf: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-89990b0a95f5.min.js",
        urban: "//assets.adobedtm.com/323c24ea560f/a9ff8797aaf9/launch-1e4fb715a4f8.min.js",
        sale: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-89990b0a95f5.min.js"
      },
      evergage: "//cdn.evgnet.com/beacon/arcteryx/sandboxospx/scripts/evergage.min.js"
    },
    stage: {
      launch: {
        outdoor: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-b4df60b59432-staging.min.js",
        leaf: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-b4df60b59432-staging.min.js",
        urban: "//assets.adobedtm.com/323c24ea560f/a9ff8797aaf9/launch-6853f8db69db-staging.min.js",
        sale: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-b4df60b59432-staging.min.js"
      },
      evergage: "//cdn.evgnet.com/beacon/arcteryx/sandboxospx/scripts/evergage.min.js"
    },
    qa: {
      launch: {
        outdoor: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-75d11936f949-development.min.js",
        leaf: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-75d11936f949-development.min.js",
        urban: "//assets.adobedtm.com/323c24ea560f/a9ff8797aaf9/launch-5ff4e649f6d5-development.min.js",
        sale: "//assets.adobedtm.com/323c24ea560f/87ce3a19f54b/launch-75d11936f949-development.min.js"
      },
      evergage: "//cdn.evgnet.com/beacon/arcteryx/qaospx/scripts/evergage.min.js"
    }
  };

  const _excluded$2 = ["env"];

  // env: production | preprod | stage | qa
  const TagManagerScript = _ref => {
    let {
        env
      } = _ref,
      props = _objectWithoutProperties(_ref, _excluded$2);
    const [disableAdobeLaunch, setDisableAdobeLaunch] = React.useState(true);

    // Get URL query param in useEffect, which will only execute on client side
    // This resolves the error of "window not defined" during SSR
    React.useEffect(() => {
      const getQueryParam = queryParam => {
        const searchParams = new URLSearchParams(window.location.search);
        return searchParams.get(queryParam);
      };
      setDisableAdobeLaunch(getQueryParam("disableAdobeLaunch") === "1");
    }, []);
    const {
      market
    } = React.useContext(componentsContexts.SiteContext);
    const url = config[env] && config[env].launch[market] ? config[env].launch[market] : null;
    return !disableAdobeLaunch ? /*#__PURE__*/React__default["default"].createElement(Script, _extends({
      url: url
    }, props)) : null;
  };

  const _excluded$1 = ["uiBaseUrl", "scriptPrefix", "env", "loadFooter"];

  // env: production | preprod | stage | qa
  // scriptPrefix: Thhe first 10 characters for the shasum value for BITBUCKET_BRANCH (e.g. "c83c2bebc8/")
  const UIComponentsScript = _ref => {
    let {
        uiBaseUrl = "",
        scriptPrefix = "",
        env,
        loadFooter = true
      } = _ref,
      props = _objectWithoutProperties(_ref, _excluded$1);
    const {
      market,
      country,
      language
    } = React.useContext(componentsContexts.SiteContext);
    const headerUrl = `${uiBaseUrl}/${country}/${language}/${scriptPrefix}${market}Header.js`;
    const footerUrl = `${uiBaseUrl}/${country}/${language}/${scriptPrefix}${market}Footer.js`;
    const marketUCFirst = market.charAt(0).toUpperCase() + market.slice(1);
    const localeSwitcherUrl = `${uiBaseUrl}/${country}/${language}/${scriptPrefix}localeSwitcher${marketUCFirst}.js`;
    return /*#__PURE__*/React__default["default"].createElement(React__default["default"].Fragment, null, /*#__PURE__*/React__default["default"].createElement(Script, _extends({
      url: headerUrl
    }, props)), loadFooter ? /*#__PURE__*/React__default["default"].createElement(Script, _extends({
      url: footerUrl,
      async: false,
      defer: true
    }, props)) : null, market !== "leaf" ? /*#__PURE__*/React__default["default"].createElement(Script, _extends({
      url: localeSwitcherUrl,
      async: false,
      defer: true
    }, props)) : null);
  };

  const HeaderFragment = _ref => {
    let {
      className,
      env = "production",
      loadScript = true,
      loadFooter = true,
      fdp = true,
      disableSearchCss = true,
      uiBaseUrl = "",
      scriptPrefix = ""
    } = _ref;
    const {
      market,
      country,
      language
    } = React.useContext(componentsContexts.SiteContext);
    const header = /*#__PURE__*/React__default["default"].createElement(`arcteryx-${market}-header`, {
      class: className,
      market,
      country,
      language,
      fdp,
      disableSearchCss,
      id: "arcteryx-header"
    });
    const localeSwitcher = /*#__PURE__*/React__default["default"].createElement("arcteryx-locale-switcher", {
      country,
      language
    });
    return /*#__PURE__*/React__default["default"].createElement(React__default["default"].Fragment, null, loadScript && /*#__PURE__*/React__default["default"].createElement(UIComponentsScript, {
      uiBaseUrl: uiBaseUrl,
      env: env,
      loadFooter: loadFooter,
      scriptPrefix: scriptPrefix
    }), header, market !== "leaf" ? localeSwitcher : null);
  };
  const StyledHeader = styled__default["default"](HeaderFragment)`
  display: block;
  min-height: ${props => props.mobileHeight};
  @media (min-width: ${props => props.breakPoint}) {
    min-height: ${props => props.desktopHeight};
  }
`;
  const headerData = {
    outdoor: {
      mobileHeight: "119.5px",
      desktopHeight: "184.5px",
      breakPoint: `${componentsTypography.ScreenSize.md}px`
    },
    sale: {
      mobileHeight: "92.5px",
      desktopHeight: "142.5px",
      breakPoint: "900px"
    },
    leaf: {
      mobileHeight: "50px",
      desktopHeight: "100px",
      breakPoint: `${componentsTypography.ScreenSize.md}px`
    }
  };
  const UIHeader = props => {
    const {
      market
    } = React.useContext(componentsContexts.SiteContext);
    const headerSizeData = React.useMemo(() => {
      // Use Leaf as a fallback since it has the smalest height
      const theMarket = market || "leaf";
      return headerData[theMarket];
    }, [market]);
    return /*#__PURE__*/React__default["default"].createElement(StyledHeader, _extends({
      desktopHeight: headerSizeData.desktopHeight,
      mobileHeight: headerSizeData.mobileHeight,
      breakPoint: headerSizeData.breakPoint
    }, props));
  };
  UIHeader.propTypes = {
    className: PropTypes__default["default"].any,
    env: PropTypes__default["default"].string,
    loadScript: PropTypes__default["default"].bool,
    loadFooter: PropTypes__default["default"].bool,
    fdp: PropTypes__default["default"].bool,
    disableSearchCss: PropTypes__default["default"].bool,
    uiBaseUrl: PropTypes__default["default"].string,
    scriptPrefix: PropTypes__default["default"].string
  };

  const _excluded = ["uiBaseUrl", "scriptPrefix", "env", "loadScript"];
  const UIFooterElement = _ref => {
    let {
      market,
      country,
      language
    } = _ref;
    return /*#__PURE__*/React__default["default"].createElement(`arcteryx-${market}-footer`, {
      country,
      language,
      id: "arcteryx-footer"
    });
  };

  // scriptPrefix: Thhe first 10 characters for the shasum value for BITBUCKET_BRANCH (e.g. "c83c2bebc8/")
  const UIFooter = _ref2 => {
    let {
        uiBaseUrl = "",
        scriptPrefix = "",
        env,
        loadScript = false
      } = _ref2,
      props = _objectWithoutProperties(_ref2, _excluded);
    const {
      market,
      country,
      language
    } = React.useContext(componentsContexts.SiteContext);
    const footerUrl = `${uiBaseUrl}/${country}/${language}/${scriptPrefix}${market}Footer.js`;
    return /*#__PURE__*/React__default["default"].createElement(React__default["default"].Fragment, null, loadScript ? /*#__PURE__*/React__default["default"].createElement(Script, _extends({
      url: footerUrl,
      async: false,
      defer: true
    }, props)) : null, /*#__PURE__*/React__default["default"].createElement(UIFooterElement, {
      market: market,
      country: country,
      language: language
    }));
  };
  UIFooter.propTypes = {
    env: PropTypes__default["default"].oneOf(["production", "stage", "qa"]),
    loadScript: PropTypes__default["default"].bool
  };

  /** Safe distance to start loading the footer script in.
   * Unless the user is scrolling very fast the footer will load before it appears on screen
   */
  const UI_LAZY_FOOTER_DEFAULT_ROOT_MARGIN = "500px";
  const FooterWrapperDiv = styled__default["default"].div`
  min-height: ${props => props.mobileHeight};
  @media (min-width: ${props => props.breakPoint}) {
    min-height: ${props => props.desktopHeight};
  }
`;

  /** Reserve space for the UIFooter element so that it doesn't cause scrollbar jank if it is lazy-loaded
    min-height determined by smallest height of Footer on different viewports for safety
  */
  const footerData = {
    outdoor: {
      appViewHeight: "0px",
      mobileHeight: "814px",
      desktopHeight: "680px",
      breakPoint: `${componentsTypography.ScreenSize.md}px`
    },
    sale: {
      appViewHeight: "800px",
      mobileHeight: "800px",
      desktopHeight: "713px",
      breakPoint: "900px"
    },
    leaf: {
      appViewHeight: "260px",
      mobileHeight: "260px",
      desktopHeight: "260px",
      breakPoint: `${componentsTypography.ScreenSize.md}px`
    }
  };

  /**
   * @param {string} rootMargin how much space from the screen to when the window meets
   * @param {boolean} appWebview user agent for mobile app
   */
  const LazyUIFooter = _ref => {
    let {
      deployEnv = "qa",
      rootMargin = UI_LAZY_FOOTER_DEFAULT_ROOT_MARGIN,
      uiBaseUrl = "",
      scriptPrefix = "",
      appWebview = false
    } = _ref;
    const {
      market
    } = React.useContext(componentsContexts.SiteContext);
    const {
      ref,
      isOnScreen
    } = componentsHooks.useIsOnScreen({
      rootMargin
    });
    const minHeight = React.useMemo(() => {
      const theMarket = market || "leaf";
      return footerData[theMarket];
    }, [market]);
    const wrapperMobileHeight = appWebview ? minHeight.appViewHeight : minHeight.mobileHeight;
    return /*#__PURE__*/React__default["default"].createElement(FooterWrapperDiv, {
      ref: ref,
      className: "qa--footer",
      "data-testid": "qa--footer",
      mobileHeight: wrapperMobileHeight,
      desktopHeight: minHeight.desktopHeight,
      breakPoint: minHeight.breakPoint
    }, isOnScreen ? /*#__PURE__*/React__default["default"].createElement(UIFooter, {
      uiBaseUrl: uiBaseUrl,
      scriptPrefix: scriptPrefix,
      loadScript: true,
      env: deployEnv
    }) : null);
  };
  LazyUIFooter.propTypes = {
    deployEnv: PropTypes__default["default"].oneOf(["production", "stage", "qa"]),
    rootMargin: PropTypes__default["default"].string,
    uiBaseUrl: PropTypes__default["default"].string,
    scriptPrefix: PropTypes__default["default"].string,
    appWebview: PropTypes__default["default"].bool
  };

  /** Loads header and locale switcher scripts
   * Lazy loads footer script
   */
  const HeaderFooterLayout = _ref => {
    let {
      children,
      deployEnv = "qa",
      headerProps,
      footerProps,
      uiBaseUrl = "",
      scriptPrefix = ""
    } = _ref;
    return /*#__PURE__*/React__default["default"].createElement(React__default["default"].Fragment, null, /*#__PURE__*/React__default["default"].createElement(UIHeader, _extends({
      uiBaseUrl: uiBaseUrl,
      scriptPrefix: scriptPrefix,
      env: deployEnv,
      loadFooter: false
    }, headerProps)), children, /*#__PURE__*/React__default["default"].createElement(LazyUIFooter, _extends({
      uiBaseUrl: uiBaseUrl,
      scriptPrefix: scriptPrefix,
      deployEnv: deployEnv
    }, footerProps)));
  };
  HeaderFooterLayout.propTypes = {
    children: PropTypes__default["default"].any,
    deployEnv: PropTypes__default["default"].oneOf(["production", "stage", "qa"]),
    headerProps: PropTypes__default["default"].shape({
      className: PropTypes__default["default"].any,
      fdp: PropTypes__default["default"].bool,
      disableSearchCss: PropTypes__default["default"].bool
    }),
    footerProps: PropTypes__default["default"].shape({
      rootMargin: PropTypes__default["default"].string,
      appWebview: PropTypes__default["default"].bool
    }),
    scriptPrefix: PropTypes__default["default"].string
  };

  const COOKIE_SCRIPT_PREFIX_KEY = "testScriptPrefix";

  exports.COOKIE_SCRIPT_PREFIX_KEY = COOKIE_SCRIPT_PREFIX_KEY;
  exports.HeaderFooterLayout = HeaderFooterLayout;
  exports.LazyUIFooter = LazyUIFooter;
  exports.TagManagerScript = TagManagerScript;
  exports.UIComponentsScript = UIComponentsScript;
  exports.UIFooter = UIFooter;
  exports.UIHeader = UIHeader;
  exports.UI_LAZY_FOOTER_DEFAULT_ROOT_MARGIN = UI_LAZY_FOOTER_DEFAULT_ROOT_MARGIN;
  exports.config = config;

  Object.defineProperty(exports, '__esModule', { value: true });

}));
