// @arcteryx/components-breadcrumb/dist/cjs/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var styled = require('styled-components');
var PropTypes = require('prop-types');
var componentsTypography = require('@arcteryx/components-typography');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var styled__default = /*#__PURE__*/_interopDefaultLegacy(styled);
var PropTypes__default = /*#__PURE__*/_interopDefaultLegacy(PropTypes);

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;
  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }
  return target;
}
function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;
  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }
  return target;
}
function _taggedTemplateLiteral(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }
  return Object.freeze(Object.defineProperties(strings, {
    raw: {
      value: Object.freeze(raw)
    }
  }));
}

var _excluded = ["tag", "children"];
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6, _templateObject7;
var BreadcrumbWrapper = styled__default["default"].div(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n  display: inline-block;\n  text-align: left;\n  clear: both;\n  text-transform: capitalize;\n  width: 100%;\n  padding: 0;\n  max-width: 95.333rem;\n  margin: var(--space-green) auto;\n  background: transparent;\n  @media screen and (min-width: 931px) {\n    text-align: left;\n  }\n"])));
var BreadcrumbListStyled = styled__default["default"].h1(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n  font-weight: normal;\n  font-size: 1rem;\n  line-height: 1.5rem;\n"])));
var BreadcrumbList = function BreadcrumbList(_ref) {
  var tag = _ref.tag,
    children = _ref.children,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/React__default["default"].createElement(BreadcrumbListStyled, _extends({}, props, {
    as: tag
  }), children);
};
var BreadcrumbListItem = styled__default["default"].span(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n  display: inline;\n  text-transform: capitalize;\n\n  font-size: 1rem;\n  line-height: 1rem;\n\n  &:first-of-type {\n    span {\n      padding-left: 0;\n    }\n  }\n\n  &:nth-last-child(2) {\n    a:after {\n      content: \"\\203A\";\n      display: inline-block;\n    }\n  }\n\n  span,\n  a:after {\n    color: ", ";\n  }\n\n  &:last-of-type {\n    display: block;\n\n    .inline & {\n      display: inline;\n    }\n\n    span,\n    a span {\n      display: inline-block;\n      width: auto;\n      text-decoration: none;\n      padding: 0.5rem 0 0;\n      text-transform: uppercase;\n      font-family: urw-din, sans-serif;\n      color: ", ";\n      font-size: 1.6rem;\n      line-height: 1.466rem;\n      .inline & {\n        display: inline-block;\n        text-transform: capitalize;\n        font-size: 1rem;\n        padding: 0 0 0 0.5rem;\n      }\n      .inline & {\n        @media (max-width: 930px) {\n          display: none;\n        }\n      }\n    }\n\n    a:after {\n      content: none;\n    }\n  }\n"], ["\n  display: inline;\n  text-transform: capitalize;\n\n  font-size: 1rem;\n  line-height: 1rem;\n\n  &:first-of-type {\n    span {\n      padding-left: 0;\n    }\n  }\n\n  &:nth-last-child(2) {\n    a:after {\n      content: \"\\\\203A\";\n      display: inline-block;\n    }\n  }\n\n  span,\n  a:after {\n    color: ", ";\n  }\n\n  &:last-of-type {\n    display: block;\n\n    .inline & {\n      display: inline;\n    }\n\n    span,\n    a span {\n      display: inline-block;\n      width: auto;\n      text-decoration: none;\n      padding: 0.5rem 0 0;\n      text-transform: uppercase;\n      font-family: urw-din, sans-serif;\n      color: ", ";\n      font-size: 1.6rem;\n      line-height: 1.466rem;\n      .inline & {\n        display: inline-block;\n        text-transform: capitalize;\n        font-size: 1rem;\n        padding: 0 0 0 0.5rem;\n      }\n      .inline & {\n        @media (max-width: 930px) {\n          display: none;\n        }\n      }\n    }\n\n    a:after {\n      content: none;\n    }\n  }\n"])), function (props) {
  return props.colour ? "".concat(props.colour) : "#fff";
}, function (props) {
  return props.colour ? "".concat(props.colour) : "#fff";
});
var BreadcrumbListItemSpan = styled__default["default"].span(_templateObject4 || (_templateObject4 = _taggedTemplateLiteral(["\n  text-transform: capitalize;\n  text-shadow: none;\n  font-style: normal;\n  padding: 0 0.5rem;\n  text-decoration: underline;\n  font-family: \"urw-din\", sans-serif;\n  font-size: 1rem;\n  line-height: 1.4rem;\n"])));
var BreadcrumbListLink = styled__default["default"].a(_templateObject5 || (_templateObject5 = _taggedTemplateLiteral(["\n  &:after {\n    content: \"\\203A\";\n    display: inline-block;\n  }\n"], ["\n  &:after {\n    content: \"\\\\203A\";\n    display: inline-block;\n  }\n"])));
var BreadcrumbLabel = styled__default["default"](componentsTypography.H1)(_templateObject6 || (_templateObject6 = _taggedTemplateLiteral(["\n  text-decoration: none;\n  margin-bottom: 0;\n  color: ", ";\n  font-weight: 700;\n  line-height: 2.4rem;\n  font-size: 2rem;\n  white-space: nowrap;\n  ", "\n"])), function (props) {
  return props.colour ? "".concat(props.colour) : "#fff";
}, componentsTypography.Breakpoint("md")(_templateObject7 || (_templateObject7 = _taggedTemplateLiteral(["\n    font-size: 2.5rem;\n    line-height: 2.9rem;\n    white-space: pre-wrap;\n  "]))));
var isIframe = function isIframe() {
  if (typeof window === "undefined") {
    return false;
  }
  return window.self !== window.top;
};
var Breadcrumb = function Breadcrumb(props) {
  if (isIframe() && !props.overrideHideInIframe && !props.isStorybook || !props.breadcrumbs) {
    return "";
  }
  return /*#__PURE__*/React__default["default"].createElement(BreadcrumbWrapper, {
    id: "breadcrumb",
    itemScope: true,
    itemType: "http://schema.org/WebPage",
    className: props.theme
  }, /*#__PURE__*/React__default["default"].createElement("nav", {
    "aria-label": "breadcrumbs"
  }, /*#__PURE__*/React__default["default"].createElement("p", {
    id: "breadcrumbLabel",
    style: {
      display: "none"
    }
  }, "You are here:"), /*#__PURE__*/React__default["default"].createElement(BreadcrumbList, {
    tag: props.tag,
    "aria-labelledby": "breadcrumbLabel",
    itemScope: true,
    itemType: "http://schema.org/BreadcrumbList"
  }, props.breadcrumbs.map(function (breadcrumb, index) {
    var isLastBreadcrumb = props.breadcrumbs.length - 1 === index;
    var breadcrumbLabel;
    if (isLastBreadcrumb) {
      var _props$theme;
      breadcrumbLabel = /*#__PURE__*/React__default["default"].createElement(BreadcrumbListItemSpan, {
        itemProp: "name"
      }, (_props$theme = props.theme) !== null && _props$theme !== void 0 && _props$theme.includes("inline") ? breadcrumb.label : /*#__PURE__*/React__default["default"].createElement(BreadcrumbLabel, {
        colour: props.colour
      }, breadcrumb.label));
    } else {
      breadcrumbLabel = /*#__PURE__*/React__default["default"].createElement(BreadcrumbListLink, {
        itemProp: "item",
        href: breadcrumb.url
      }, /*#__PURE__*/React__default["default"].createElement(BreadcrumbListItemSpan, {
        itemProp: "name"
      }, breadcrumb.label));
    }
    return /*#__PURE__*/React__default["default"].createElement(BreadcrumbListItem, {
      itemProp: "itemListElement",
      itemScope: true,
      itemType: "http://schema.org/ListItem",
      key: index,
      colour: props.colour
    }, breadcrumbLabel, /*#__PURE__*/React__default["default"].createElement("meta", {
      itemProp: "position",
      content: "@itempropPosition"
    }));
  }))));
};
Breadcrumb.defaultProps = {
  colour: "#000",
  tag: "h1"
};
BreadcrumbListItem.defaultProps = {
  colour: Breadcrumb.defaultProps.colour,
  tag: Breadcrumb.defaultProps.tag
};
Breadcrumb.propTypes = {
  breadcrumbs: PropTypes__default["default"].arrayOf(PropTypes__default["default"].shape({
    label: PropTypes__default["default"].string.isRequired,
    url: PropTypes__default["default"].string.isRequired
  })).isRequired,
  colour: PropTypes__default["default"].string,
  className: PropTypes__default["default"].string,
  tag: PropTypes__default["default"].string,
  theme: PropTypes__default["default"].string,
  overrideHideInIframe: PropTypes__default["default"].bool,
  isStorybook: PropTypes__default["default"].bool // required to show Breadcrumbs in Storybook since it's loaded in an iFrame
};

exports.Breadcrumb = Breadcrumb;
