// @arcteryx/design-system/src/components/molecules/ImageTextLinkBox/ImageTextLinkBox.tsx
import React from "react";
import styled from "styled-components";
import { FullWidthImage } from "../fullWidthImage";
import { splitLabel } from "../../../utils/libs";
import { type ImageTextLinkBoxProps } from "./ImageTextLinkBox.types";

const StyledImageTextLinkBox = styled.a`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-decoration: none;
  color: inherit;
  position: relative;
  width: 100%;
`;

const ImageOverlay = styled.span`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: white;
  font-size: 2rem;
  line-height: 2rem;
  font-weight: normal;
  text-transform: uppercase;
  font-family: "urw-din-semi-condensed", sans-serif;
  text-align: center;
  text-shadow: 2px 2px 10px rgba(0, 0, 0, 0.4);
`;

const FirstPart = styled.span`
  font-weight: 700;
  padding-right: 0.33rem;
`;

export const ImageTextLinkBox: React.FC<ImageTextLinkBoxProps> = ({
  image,
  label,
  url,
  target = "_blank",
}) => {
  const [firstPart, restLabel] = splitLabel(label);
  return (
    <StyledImageTextLinkBox
      href={url}
      target={target}
      data-testid="ImageTextLinkBox"
    >
      <FullWidthImage
        image={image?.image}
        sourceElementsArray={image?.sourceElementsArray}
      />
      <ImageOverlay>
        <FirstPart>{firstPart}</FirstPart>
        <span>{restLabel}</span>
      </ImageOverlay>
    </StyledImageTextLinkBox>
  );
};
